---
title: "Tutorial for MotrpacRatTraining6moData R package"
author: "Nicole R. Gay, David Amar"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_document:
    toc: true
    code_folding: show 
    toc_depth: 4
    toc_float:
      collapsed: false
      smooth_scroll: false
      
vignette: >
  %\VignetteIndexEntry{Tutorial for MotrpacRatTraining6moData R package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction 

> This vignette is under construction. If something is unclear or obviously missing, 
please submit an issue [here](https://github.com/MoTrPAC/MotrpacRatTraining6moData/issues). 

### About this package 
This package provides convenient access to the processed data and downstream
analysis results presented in the main paper for the first 
large-scale multi-omic multi-tissue endurance exercise training study conducted 
in young adult rats by the Molecular Transducers of Physical Activity Consortium 
(MoTrPAC). *A [bioRxiv](https://www.biorxiv.org/) link to the corresponding 
preprint will be added shortly.* 

While the data in this package can be used by themselves, the 
[MotrpacRatTraining6mo](https://github.com/MoTrPAC/MotrpacRatTraining6mo)
R package relies heavily on this package and provides many functions to help
retrieve and explore the data. *A link to the MotrpacRatTraining6mo vignette
will be added shortly.* 

### About MoTrPAC 
MoTrPAC is a national research consortium designed to discover and perform 
preliminary characterization of the range of molecular transducers (the 
"molecular map") that underlie the effects of physical activity in humans. 
The program's goal is to study the molecular changes that occur during and after 
exercise and ultimately to advance the understanding of how physical activity 
improves and preserves health. The six-year program is the largest targeted NIH 
investment of funds into the mechanisms of how physical activity improves health 
and prevents disease. See [motrpac.org](https://www.motrpac.org/) and 
[motrpac-data.org](https://motrpac-data.org/) for more details. 

### Installation
Install this package with `devtools`:
```r
if (!require("devtools", quietly = TRUE)){
  install.packages("devtools")
}
devtools::install_github("MoTrPAC/MotrpacRatTraining6moData")
```

The output for a successful installation looks something like this. 
Note that the `*** moving datasets to lazyload DB` step takes the longest (~5 minutes):
```
Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
✓  checking for file ‘.../MoTrPAC-MotrpacRatTraining6moData-1c6478a/DESCRIPTION’ ...
─  preparing ‘MotrpacRatTraining6moData’:
✓  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘MotrpacRatTraining6moData_1.3.2.tar.gz’ (1.3s)
   
* installing *source* package ‘MotrpacRatTraining6moData’ ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (MotrpacRatTraining6moData)
```

#### Troubleshooting
If you get this error:  
```
Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD' failed
```
Try extending the timeout:  
```r
options(timeout=1e5)
devtools::install_github("MoTrPAC/MotrpacRatTraining6moData")
```

If you get this error after extending the `timeout`:  
```
Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD' failed
Error in `action()`:
! `class` is absent but must be supplied.
Run `rlang::last_error()` to see where the error occurred.
```
...this seems to be an intermittent issue seen only on Mac, not Linux or Windows. 
This was resolved by [installing the newest version of R](https://cran.rstudio.com/). 

#### Last resort
If you can't get `devtools::install_github("MoTrPAC/MotrpacRatTraining6moData")` to work, try this:  

1. Go to <https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD>, which will automatically start downloading this repository in a tarball 
2. Install the package from source: 
   ```r
   install.packages("~/Downloads/MoTrPAC-MotrpacRatTraining6moData-0729e2e.tar.gz", 
     repos = NULL, 
     type = "source")
   library(MotrpacRatTraining6moData)
   ```

Once we install this package, we can load the library. 
```{r setup}
library(MotrpacRatTraining6moData)
```

## Study design 
Details of the experimental design can be found in the supplementary methods 
of the bioRxiv preprint (TODO: add link). Briefly, 6-month-old young adult rats 
were subjected to progressive endurance exercise training
for 1, 2, 4, or 8 weeks, with tissues collected 48 hours after the last training bout. 
Sex-matched sedentary, untrained rats were used as controls. Whole blood, plasma, 
and 18 solid tissues were analyzed using genomics, proteomics, metabolomics, 
and protein immunoassay technologies, with most assays performed in a subset of these tissues. 
Depending on the assay, between 3 and 5 replicates per sex per time point were
analyzed. 

## Tissue and assay abbreviations  
It is important to be aware of the tissue and assay abbreviations because they are used
to name data objects and define arguments for many functions.  

### Tissues
* ADRNL: adrenal gland  
* BAT: brown adipose tissue  
* BLOOD: whole blood  
* COLON: colon  
* CORTEX: cerebral cortex  
* HEART: heart  
* HIPPOC: hippocampus  
* HYPOTH: hypothalamus  
* KIDNEY: kidney  
* LIVER: liver  
* LUNG: lung  
* OVARY: ovaries (female gonads)  
* PLASMA: plasma from blood
* SKM-GN: gastrocnemius (leg skeletal muscle)  
* SKM-VL: vastus lateralis (leg skeletal muscle) 
* SMLINT: small intestine  
* SPLEEN: spleen  
* TESTES: testes (male gonads)  
* VENACV: vena cava  
* WAT-SC: subcutaneous white adipose tissue  

### Assays/omes
* ACETYL: acetylproteomics; protein site acetylation  
* ATAC: chromatin accessibility, ATAC-seq data  
* IMMUNO: multiplexed immunoassays (cytokines and hormones)  
* METAB: metabolomics and lipidomics  
* METHYL: DNA methylation, RRBS data   
* PHOSPHO: phosphoproteomics; protein site phosphorylation  
* PROT: global proteomics; protein abundance  
* TRNSCRPT: transcriptomics, RNA-Seq data  
* UBIQ: ubiquitylome; protein site ubiquitination  

## Summary of data 
Here is a brief summary of the kinds of data included in this package: 

* Assay, tissue, sex, and training group abbreviations, codes, colors, and order used in plots
* Phenotypic data, `PHENO`  
* Mapping between various feature identifiers, i.e., `FEATURE_TO_GENE`, `RAT_TO_HUMAN_GENE`  
* Ome-specific feature annotation, i.e., `METAB_FEATURE_ID_MAP`, `METHYL_FEATURE_ANNOT` (GCP only), `ATAC_FEATURE_ANNOT` (GCP only)
* Ome-specific sample-level metadata, i.e., `TRNSCRPT_META`, `ATAC_META`, `METHYL_META`, `IMMUNO_META`, `PHOSPHO_META`, `PROT_META`, `ACETYL_META`, `UBIQ_META`  
* Raw counts for RNA-Seq (TRNSCRPT), ATAC-Seq (ATAC), and RRBS (METHYL) data, 
e.g., `TRNSCRPT_LIVER_RAW_COUNTS`.
Note that epigenetic data (ATAC and METHYL) must be downloaded from Google Cloud Storage.  
* Normalized sample-level data, e.g., `TRNSCRPT_SKMGN_NORM_DATA`   
* Differential analysis results, e.g., `HEART_PROT_DA` (see more details below)  
* Sample outliers excluded from differential analysis, `OUTLIERS`  
* Table of training-regulated features at 5% FDR, `TRAINING_REGULATED_FEATURES`  
* Bayesian graphical analysis inputs and results (see more details below)  
* Pathway enrichment of main graphical clusters, `GRAPH_PW_ENRICH`  

<!-- Start examples here -->

## Getting help 
For questions, bug reporting, and data requests for this package, please 
[submit a new issue](https://github.com/MoTrPAC/MotrpacRatTraining6moData/issues) 
and include as many details as possible. 

If the concern is related to functions provided in the 
[MotrpacRatTraining6mo](https://github.com/MoTrPAC/MotrpacRatTraining6mo)
package, please submit an issue 
[here](https://github.com/MoTrPAC/MotrpacRatTraining6mo/issues) instead. 

## Acknowledgements 
MoTrPAC is supported by the National Institutes of Health (NIH) Common
Fund through cooperative agreements managed by the National Institute of Diabetes and
Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal
Diseases (NIAMS), and National Institute on Aging (NIA). 
