<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="ATAC-seq feature annotation"><title>ATAC-seq feature annotation — ATAC_FEATURE_ANNOT • MotrpacRatTraining6moData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ATAC-seq feature annotation — ATAC_FEATURE_ANNOT"><meta property="og:description" content="ATAC-seq feature annotation"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6moData.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>ATAC-seq feature annotation</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>ATAC_FEATURE_ANNOT.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>ATAC-seq feature annotation</p>
    </div>


    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with 1209773 rows and 16 variables:</p><dl><dt><code>assay</code></dt>
<dd><p>character, assay abbreviation, one of <a href="ASSAY_ABBREV.html">ASSAY_ABBREV</a></p></dd>

<dt><code>assay_code</code></dt>
<dd><p>character, assay code used in data release. See <a href="https://rdrr.io/pkg/MotrpacBicQC/man/assay_codes.html" class="external-link">MotrpacBicQC::assay_codes</a>.</p></dd>

<dt><code>feature_ID</code></dt>
<dd><p>character, MoTrPAC feature identifier</p></dd>

<dt><code>chrom</code></dt>
<dd><p>character, chromosome: 1-20, X, or Y</p></dd>

<dt><code>start</code></dt>
<dd><p>double, base pair of feature start</p></dd>

<dt><code>end</code></dt>
<dd><p>double, base bair of feature end</p></dd>

<dt><code>width</code></dt>
<dd><p>integer, width of feature in base pairs</p></dd>

<dt><code>chipseeker_annotation</code></dt>
<dd><p>character, annotation from <code><a href="https://rdrr.io/pkg/ChIPseeker/man/annotatePeak.html" class="external-link">ChIPseeker::annotatePeak()</a></code></p></dd>

<dt><code>custom_annotation</code></dt>
<dd><p>character, a version of the ChIPseeker annotations with many corrections. Values include:
"Distal Intergenic", "Promoter (&lt;=1kb)", "Exon", "Promoter (1-2kb)", "Downstream (&lt;5kb)", "Upstream (&lt;5kb)", "5' UTR",
"Intron", "3' UTR", "Overlaps Gene", where "Overlaps Gene" means the feature has a non-zero overlap with
either the start or end of the gene but was not otherwise asssigned an annotation.</p></dd>

<dt><code>distanceToTSS</code></dt>
<dd><p>double, minimum distance from one end of the feature to the transcription start site.</p></dd>

<dt><code>relationship_to_gene</code></dt>
<dd><p>double, distance from the closest edge
of the feature to the start or end of the closest gene, whichever is closer.
A value of 0 means there is non-zero overlap between the feature and the gene.
A negative value means the feature is upstream of <code>geneStart</code>.
A a positive value means the feature is downstream of <code>geneEnd</code>.
Note that <code>geneStart</code> and <code>geneEnd</code> are strand-agnostic, i.e. <code>geneStart</code>
is always less than <code>geneEnd</code>, even if the gene is on the negative strand (<code>geneStrand == 2</code>).</p></dd>

<dt><code>ensembl_gene</code></dt>
<dd><p>character, Ensembl gene ID from release 96 of the Rattus norvegicus gene annotation</p></dd>

<dt><code>geneStart</code></dt>
<dd><p>integer, base pair start of gene; strand-agnostic, meaning <code>geneStart</code> is always less than <code>geneEnd</code></p></dd>

<dt><code>geneEnd</code></dt>
<dd><p>integer, base pair end of gene; strand-agnostic, meaning <code>geneStart</code> is always less than <code>geneEnd</code></p></dd>

<dt><code>geneLength</code></dt>
<dd><p>integer, length of gene in base pairs</p></dd>

<dt><code>geneStrand</code></dt>
<dd><p>integer, 1 (forward strand) or 2 (reverse strand)</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>ATAC feature annotation is only available via download from Google Cloud Storage:
<a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_FEATURE_ANNOT.rda" class="external-link">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_FEATURE_ANNOT.rda</a>.
You can use <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_atac_feature_annotation.html" class="external-link">MotrpacRatTraining6mo::load_atac_feature_annotation()</a></code> to download and return this file.</p>
<p>This table was generated using <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/get_peak_annotations.html" class="external-link">MotrpacRatTraining6mo::get_peak_annotations()</a></code>.
<code>relationship_to_gene</code> is the shortest distance between the feature and the start or end of the closest gene.
It is 0 if the feature has any overlap with the gene.
<code>custom_annotation</code> fixes many issues with the <code>ChIPseeker</code> annotation (v1.22.1).</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

