<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="ATAC-seq experimental and quantification QC metrics for chromatin accessibility (ATAC) data"><title>ATAC-seq metadata and QC — ATAC_META • MotrpacRatTraining6moData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ATAC-seq metadata and QC — ATAC_META"><meta property="og:description" content="ATAC-seq experimental and quantification QC metrics for chromatin accessibility (ATAC) data"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6moData.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>ATAC-seq metadata and QC</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>ATAC_META.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>ATAC-seq experimental and quantification QC metrics for chromatin accessibility (ATAC) data</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">ATAC_META</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with 520 rows and 106 variables:</p><dl><dt><code>viallabel</code></dt>
<dd><p>character, sample identifier</p></dd>

<dt><code>general.description</code></dt>
<dd><p>character, pipeline workflow description</p></dd>

<dt><code>replicate</code></dt>
<dd><p>character, replicate in the pipeline workflow</p></dd>

<dt><code>general.date</code></dt>
<dd><p>double, date the pipeline workflow was run</p></dd>

<dt><code>general.title</code></dt>
<dd><p>character, pipeline workflow title</p></dd>

<dt><code>general.pipeline_ver</code></dt>
<dd><p>character, ENCODE ATAC-seq pipeline version</p></dd>

<dt><code>general.pipeline_type</code></dt>
<dd><p>character, "atac"</p></dd>

<dt><code>general.genome</code></dt>
<dd><p>character, reference genome</p></dd>

<dt><code>general.aligner</code></dt>
<dd><p>character, read aligner</p></dd>

<dt><code>general.peak_caller</code></dt>
<dd><p>character, peak caller</p></dd>

<dt><code>general.seq_endedness.paired_end</code></dt>
<dd><p>logical, are the reads paired-ended</p></dd>

<dt><code>replication.num_peaks.num_peaks</code></dt>
<dd><p>integer, number of replication peaks (max 300000)</p></dd>

<dt><code>peak_stat.peak_region_size.min_size</code></dt>
<dd><p>integer, minimum peak width</p></dd>

<dt><code>peak_stat.peak_region_size.25_pct</code></dt>
<dd><p>integer, 25th percentile of peak width</p></dd>

<dt><code>peak_stat.peak_region_size.50_pct</code></dt>
<dd><p>integer, 50th percentile of peak width</p></dd>

<dt><code>peak_stat.peak_region_size.75_pct</code></dt>
<dd><p>integer, 75th percentile of peak width</p></dd>

<dt><code>peak_stat.peak_region_size.max_size</code></dt>
<dd><p>integer, max peak width</p></dd>

<dt><code>peak_stat.peak_region_size.mean</code></dt>
<dd><p>double, mean peak width</p></dd>

<dt><code>peak_enrich.frac_reads_in_peaks.macs2.frip</code></dt>
<dd><p>double, fraction of reads in MACS2 peaks</p></dd>

<dt><code>align.samstat.total_reads</code></dt>
<dd><p>integer, total number of alignments, including multimappers</p></dd>

<dt><code>align.samstat.mapped_reads</code></dt>
<dd><p>integer, total number of mapped reads</p></dd>

<dt><code>align.samstat.pct_mapped_reads</code></dt>
<dd><p>double, percent of reads that mapped</p></dd>

<dt><code>align.samstat.paired_reads</code></dt>
<dd><p>integer, number of paired reads</p></dd>

<dt><code>align.samstat.read1</code></dt>
<dd><p>integer, number of read 1 reads</p></dd>

<dt><code>align.samstat.read2</code></dt>
<dd><p>integer, number of read 2 reads</p></dd>

<dt><code>align.samstat.properly_paired_reads</code></dt>
<dd><p>integer, number of properly paired reads</p></dd>

<dt><code>align.samstat.pct_properly_paired_reads</code></dt>
<dd><p>double, percent of reads that were properly paired</p></dd>

<dt><code>align.samstat.with_itself</code></dt>
<dd><p>integer, number of reads paired with its pair</p></dd>

<dt><code>align.samstat.singletons</code></dt>
<dd><p>integer, number of singleton reads</p></dd>

<dt><code>align.samstat.pct_singletons</code></dt>
<dd><p>double, percent of reads that were singleton</p></dd>

<dt><code>align.samstat.diff_chroms</code></dt>
<dd><p>integer</p></dd>

<dt><code>align.dup.paired_reads</code></dt>
<dd><p>integer, number of paired reads for duplication step</p></dd>

<dt><code>align.dup.paired_duplicate_reads</code></dt>
<dd><p>integer, number of duplicate paired reads</p></dd>

<dt><code>align.dup.paired_optical_duplicate_reads</code></dt>
<dd><p>integer, number of optical duplicate paired reads</p></dd>

<dt><code>align.dup.pct_duplicate_reads</code></dt>
<dd><p>double, percent of reads that are duplicate</p></dd>

<dt><code>align.frac_mito.non_mito_reads</code></dt>
<dd><p>integer, percent of reads that align to non-mitochondrial DNA</p></dd>

<dt><code>align.frac_mito.mito_reads</code></dt>
<dd><p>integer, number of reads that align to mitochondrial DNA</p></dd>

<dt><code>align.frac_mito.frac_mito_reads</code></dt>
<dd><p>double, fraction of reads that align to mitochondrial DNA</p></dd>

<dt><code>align.nodup_samstat.total_reads</code></dt>
<dd><p>integer, number of reads after applying all filters</p></dd>

<dt><code>align.nodup_samstat.mapped_reads</code></dt>
<dd><p>integer, number of mapped reads after applying all filters</p></dd>

<dt><code>align.nodup_samstat.paired_reads</code></dt>
<dd><p>integer, number of paired reads after applying all filters</p></dd>

<dt><code>align.nodup_samstat.read1</code></dt>
<dd><p>integer, number of read 1 reads after applying all filters</p></dd>

<dt><code>align.nodup_samstat.read2</code></dt>
<dd><p>integer, number of read 2 reads after applying all filters</p></dd>

<dt><code>align.nodup_samstat.properly_paired_reads</code></dt>
<dd><p>integer, number of properly paired reads after applying all filters</p></dd>

<dt><code>align.nodup_samstat.with_itself</code></dt>
<dd><p>integer, number of reads paired with its pair after applying all filters</p></dd>

<dt><code>align.frag_len_stat.frac_reads_in_nfr</code></dt>
<dd><p>double, fraction of reads in nucelosome-free region. Should be a value greater than 0.4.</p></dd>

<dt><code>align.frag_len_stat.frac_reads_in_nfr_qc_pass</code></dt>
<dd><p>logical, does <code>align.frag_len_stat.frac_reads_in_nfr</code> pass the cutoff?</p></dd>

<dt><code>align.frag_len_stat.frac_reads_in_nfr_qc_reason</code></dt>
<dd><p>character, reason for <code>align.frag_len_stat.frac_reads_in_nfr_qc_pass</code></p></dd>

<dt><code>align.frag_len_stat.nfr_over_mono_nuc_reads</code></dt>
<dd><p>double, reads in nucleosome-free-region
versus reads in mononucleosomal peak. Should be a value greater than 2.5.</p></dd>

<dt><code>align.frag_len_stat.nfr_over_mono_nuc_reads_qc_pass</code></dt>
<dd><p>logical, does <code>align.frag_len_stat.frac_reads_in_nfr_qc_pass</code> pass the cutoff?</p></dd>

<dt><code>align.frag_len_stat.nfr_over_mono_nuc_reads_qc_reason</code></dt>
<dd><p>character, reason for <code>align.frag_len_stat.nfr_over_mono_nuc_reads_qc_pass</code></p></dd>

<dt><code>align.frag_len_stat.nfr_peak_exists</code></dt>
<dd><p>logical, does a nucleosome-free peak exist?</p></dd>

<dt><code>align.frag_len_stat.mono_nuc_peak_exists</code></dt>
<dd><p>logical, does a mono-nucleosomal peak exist?</p></dd>

<dt><code>align.frag_len_stat.di_nuc_peak_exists</code></dt>
<dd><p>logical, does a di-nucleosomal peak exist?</p></dd>

<dt><code>lib_complexity.lib_complexity.total_fragments</code></dt>
<dd><p>integer, total number of fragments</p></dd>

<dt><code>lib_complexity.lib_complexity.distinct_fragments</code></dt>
<dd><p>integer, number of distinct fragments</p></dd>

<dt><code>lib_complexity.lib_complexity.positions_with_one_read</code></dt>
<dd><p>integer, number of positions with one read</p></dd>

<dt><code>lib_complexity.lib_complexity.NRF</code></dt>
<dd><p>double, non-reduandant fraction. Measure of library complexity. Ideally &gt;0.9</p></dd>

<dt><code>lib_complexity.lib_complexity.PBC1</code></dt>
<dd><p>double, PCR bottlenecking coefficient 1. Measure of library complexity. Ideally &gt;0.9</p></dd>

<dt><code>lib_complexity.lib_complexity.PBC2</code></dt>
<dd><p>double PCR bottlenecking coefficient 2. Measure of library complexity. Ideally &gt;3</p></dd>

<dt><code>align_enrich.tss_enrich.tss_enrich</code></dt>
<dd><p>double, transcription start site enrichment of peaks</p></dd>

<dt><code>2D_barcode</code></dt>
<dd><p>double, sample barcode</p></dd>

<dt><code>Tissue</code></dt>
<dd><p>character, tissue description</p></dd>

<dt><code>Species</code></dt>
<dd><p>character, species</p></dd>

<dt><code>Sample_category</code></dt>
<dd><p>character, study sample ("study") or reference standard ("ref)</p></dd>

<dt><code>GET_site</code></dt>
<dd><p>character, which Genomics, Epigenomics, and Transcriptomics (GET)
site performed the assay, "Stanford" or "MSSM" (Icahn School of Medicine at Mount Sinai)</p></dd>

<dt><code>Sample_batch</code></dt>
<dd><p>integer, numeric batch number for batch in which this sample was manually processed</p></dd>

<dt><code>Lib_adapter_1</code></dt>
<dd><p>character, Adapter sequence for read 1</p></dd>

<dt><code>Lib_adapter_2</code></dt>
<dd><p>character, Adapter sequence for read 2</p></dd>

<dt><code>Lib_index_1</code></dt>
<dd><p>character, i7 index</p></dd>

<dt><code>Lib_index_2</code></dt>
<dd><p>character, i5 index</p></dd>

<dt><code>Nuclei_extr_date</code></dt>
<dd><p>character, nuclei extraction date</p></dd>

<dt><code>Nuclei_extr_count</code></dt>
<dd><p>integer, nuclei count</p></dd>

<dt><code>Nuclei_tagmentation</code></dt>
<dd><p>integer, number of nuclei used in each tagmentation reaction</p></dd>

<dt><code>Tagmentation_date</code></dt>
<dd><p>character, tagmentation date, MM/DD/YYYY format</p></dd>

<dt><code>Tagmentation_enzyme_cat</code></dt>
<dd><p>integer, catalog number of tagmentation enzyme TDE1 (Tn5)</p></dd>

<dt><code>Tagmentation_enzyme_lot</code></dt>
<dd><p>integer, lot number of tagmentation enzyme TDE1 (Tn5)</p></dd>

<dt><code>Tagmentation_buffer_cat</code></dt>
<dd><p>integer, catalog number of tagmentation buffer</p></dd>

<dt><code>Tagmentation_buffer_lot</code></dt>
<dd><p>integer, lot number of tagmentation buffer</p></dd>

<dt><code>Tagmentation_reaction_vol</code></dt>
<dd><p>integer, volume of tagmentation (uL)</p></dd>

<dt><code>Tagmentation_purification_kit</code></dt>
<dd><p>character, purification kit</p></dd>

<dt><code>Tagmentation_purified_DNA_vol</code></dt>
<dd><p>double, volume of purified DNA (uL)</p></dd>

<dt><code>PCR_date</code></dt>
<dd><p>character, PCR date, MM/DD/YYYY format</p></dd>

<dt><code>PCR_cycle_nr</code></dt>
<dd><p>integer, number of PCR cycles</p></dd>

<dt><code>PCR_purification_beads_ul</code></dt>
<dd><p>character, volume of SPRIselect beads for lower size selection</p></dd>

<dt><code>Lib_DNA_conc</code></dt>
<dd><p>double, DNA concentration for the library (ng/uL)</p></dd>

<dt><code>Lib_DNA_molarity</code></dt>
<dd><p>double, DNA molarity of library (nM)</p></dd>

<dt><code>Lib_frag_size</code></dt>
<dd><p>integer, average library fragment size</p></dd>

<dt><code>Lib_BA_quality</code></dt>
<dd><p>integer, visual inspection of the library quality with the Bioanalyzer track (1=good, 0=bad)</p></dd>

<dt><code>Seq_DNA_molarity</code></dt>
<dd><p>double, DNA molarity for sequencing (nM)</p></dd>

<dt><code>Seq_platform</code></dt>
<dd><p>character, sequencing platform</p></dd>

<dt><code>Seq_date</code></dt>
<dd><p>integer, sequencing date, YYMMDD format</p></dd>

<dt><code>Seq_machine_ID</code></dt>
<dd><p>character, serial number of the sequencer</p></dd>

<dt><code>Seq_flowcell_ID</code></dt>
<dd><p>character, flow cell ID</p></dd>

<dt><code>Seq_flowcell_run</code></dt>
<dd><p>integer, flow cell run</p></dd>

<dt><code>Seq_flowcell_lane</code></dt>
<dd><p>character, flow cell lane</p></dd>

<dt><code>Seq_flowcell_type</code></dt>
<dd><p>character, flow cell type, e.g., S4</p></dd>

<dt><code>Seq_length</code></dt>
<dd><p>double, read length</p></dd>

<dt><code>Seq_end_type</code></dt>
<dd><p>integer, 1=single-end, 2=paired-end</p></dd>

<dt><code>total_primary_alignments</code></dt>
<dd><p>integer, number of primary alignments</p></dd>

<dt><code>pct_chrX</code></dt>
<dd><p>double, number of reads mapped to chromosome X</p></dd>

<dt><code>pct_chrY</code></dt>
<dd><p>double, number of reads mapped to chromosome Y</p></dd>

<dt><code>pct_chrM</code></dt>
<dd><p>double, number of reads mapped to chromosome M</p></dd>

<dt><code>pct_auto</code></dt>
<dd><p>double, number of reads mapped to autosomal chromosomes</p></dd>

<dt><code>pct_contig</code></dt>
<dd><p>double, number of reads mapped to contigs</p></dd>

<dt><code>Seq_batch</code></dt>
<dd><p>character, unique identifier for sequencing batch</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="source">Source<a class="anchor" aria-label="anchor" href="#source"></a></h2>
    <p><code>pass1b-06/results/epigenomics/qa-qc/motrpac_pass1b-06_epigen-atac-seq_qa-qc-metrics.csv</code></p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The <a href="https://github.com/ENCODE-DCC/atac-seq-pipeline/tree/v1.7.0" class="external-link">ENCODE ATAC-seq pipeline v1.7.0</a> was used to quantify ATAC-seq data.
Columns with a period are QC metrics from this pipeline. Note that the ENCODE pipeline reports alignments per paired-end read,
so <code>align.samstat.total_reads</code> reports the number of paired-end reads that align, which corresponds to twice the number of sequenced fragments.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

