<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Subset of FEATURE_TO_GENE that excludes non-differential epigenetic features."><title>Filtered feature-to-gene map — FEATURE_TO_GENE_FILT • MotrpacRatTraining6moData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Filtered feature-to-gene map — FEATURE_TO_GENE_FILT"><meta property="og:description" content="Subset of FEATURE_TO_GENE that excludes non-differential epigenetic features."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.6.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6moData.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Filtered feature-to-gene map</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>FEATURE_TO_GENE_FILT.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Subset of <a href="FEATURE_TO_GENE.html">FEATURE_TO_GENE</a> that excludes non-differential epigenetic features.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">FEATURE_TO_GENE_FILT</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with 241572 rows and 9 variables:</p><dl><dt><code>entrez_gene</code></dt>
<dd><p>double, Entrez gene ID</p></dd>

<dt><code>feature_ID</code></dt>
<dd><p>character, MoTrPAC feature identifier</p></dd>

<dt><code>rgd_gene</code></dt>
<dd><p>integer, RGD gene ID</p></dd>

<dt><code>gene_symbol</code></dt>
<dd><p>character, official gene symbol</p></dd>

<dt><code>old_gene_symbol</code></dt>
<dd><p>character, semicolon-separated list of deprecated or alias gene symbols</p></dd>

<dt><code>ensembl_gene</code></dt>
<dd><p>character, Ensembl gene ID</p></dd>

<dt><code>relationship_to_gene</code></dt>
<dd><p>double, for ATAC and METHYL features only. Distance
from the closest edge of the feature to the start or end of the closest gene, whichever is closer.
A value of 0 means there is non-zero overlap between the feature and the gene.
A negative value means the feature is upstream of "geneStart".
A a positive value means the feature is downstream of "geneEnd".
Note that "geneStart" and "geneEnd" are strand-agnostic, i.e. "geneStart"
is always less than "geneEnd", even if the gene is on the negative strand ("geneStrand" == 2).</p></dd>

<dt><code>custom_annotation</code></dt>
<dd><p>character, a version of the <code>ChIPseeker</code> annotations with many corrections. Values include:
"Distal Intergenic", "Promoter (&lt;=1kb)", "Exon", "Promoter (1-2kb)", "Downstream (&lt;5kb)", "Upstream (&lt;5kb)", "5' UTR", "Intron", "3' UTR", "Overlaps Gene"</p></dd>

<dt><code>kegg_id</code></dt>
<dd><p>character, KEGG ID for METAB features only. See <a href="https://rdrr.io/pkg/MotrpacBicQC/man/metabolomics_data_dictionary.html" class="external-link">MotrpacBicQC::metabolomics_data_dictionary</a> for more details.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>All proteomics feature IDs (RefSeq accessions) were mapped to gene
symbols and Entrez IDs using NCBI’s "gene2refseq" mapping files
(<a href="https://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2refseq.gz" class="external-link">https://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2refseq.gz</a>, downloaded on 2020/12/18).
Epigenomics features were mapped to the nearest gene using <code><a href="https://rdrr.io/pkg/ChIPseeker/man/annotatePeak.html" class="external-link">ChIPseeker::annotatePeak()</a></code>
with Ensembl gene annotation (Rattus norvegicus release 96).
See <a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/get_peak_annotations.html" class="external-link">MotrpacRatTraining6mo::get_peak_annotations</a> for implementation.
Gene symbols, Entrez IDs, Ensembl IDs, and RGD IDs were mapped to each
other using RGD’s rat gene annotation
(<a href="https://download.rgd.mcw.edu/data_release/RAT/GENES_RAT.txt" class="external-link">https://download.rgd.mcw.edu/data_release/RAT/GENES_RAT.txt</a>, generated on 2021/11/12).</p>
<p></p><div class="sourceCode"><pre><code><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>For <span class="fu">fast</span>(er) indexing, convert this object to a [data.table<span class="sc">::</span><span class="fu">data.table</span>()] and use </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>[data.table<span class="sc">::</span><span class="fu">setkey</span>()] to set the key to the column you are matching. </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>This dramatically improves performance. </span></code></pre><p></p></div>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

