<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Mapping between various rat and human gene identifiers for orthologs as reported by RGD (2020-01-10)."><title>Rat-to-human gene ortholog map — RAT_TO_HUMAN_GENE • MotrpacRatTraining6moData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Rat-to-human gene ortholog map — RAT_TO_HUMAN_GENE"><meta property="og:description" content="Mapping between various rat and human gene identifiers for orthologs as reported by RGD (2020-01-10)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.6.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6moData.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Rat-to-human gene ortholog map</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>RAT_TO_HUMAN_GENE.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Mapping between various rat and human gene identifiers for orthologs as reported by <a href="https://rgd.mcw.edu/" class="external-link">RGD</a> (2020-01-10).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">RAT_TO_HUMAN_GENE</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with 21461 rows and 12 variables:</p><dl><dt><code>RAT_SYMBOL</code></dt>
<dd><p>character, official rat gene symbol</p></dd>

<dt><code>NAME</code></dt>
<dd><p>character, gene name</p></dd>

<dt><code>RAT_GENE_RGD_ID</code></dt>
<dd><p>integer, rat RGD gene ID</p></dd>

<dt><code>RAT_NCBI_GENE_ID</code></dt>
<dd><p>integer, rat NCBI/Entrez gene ID</p></dd>

<dt><code>RAT_ENSEMBL_ID</code></dt>
<dd><p>character, rat Ensembl gene ID</p></dd>

<dt><code>RAT_UNIPROT_ID</code></dt>
<dd><p>character, semicolon-separated list of rat UniProt gene IDs</p></dd>

<dt><code>RAT_OLD_SYMBOL</code></dt>
<dd><p>character, semicolon-separated list of rat deprecated or alias gene symbols</p></dd>

<dt><code>HUMAN_ORTHOLOG_SYMBOL</code></dt>
<dd><p>character, human official gene symbol</p></dd>

<dt><code>HUMAN_ORTHOLOG_RGD</code></dt>
<dd><p>character, human RGD gene ID</p></dd>

<dt><code>HUMAN_ORTHOLOG_NCBI_GENE_ID</code></dt>
<dd><p>character, human NCBI/Entrez gene ID</p></dd>

<dt><code>HUMAN_ORTHOLOG_ENSEMBL_ID</code></dt>
<dd><p>character, human Entrez gene ID</p></dd>

<dt><code>HUMAN_ORTHOLOG_SOURCE</code></dt>
<dd><p>character, human ortholog source</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="source">Source<a class="anchor" aria-label="anchor" href="#source"></a></h2>
    <p></p><div class="sourceCode"><pre><code><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&lt;</span>https<span class="sc">:</span><span class="er">//</span>ftp.ebi.ac.uk<span class="sc">/</span>pub<span class="sc">/</span>databases<span class="sc">/</span>gencode<span class="sc">/</span>Gencode_human<span class="sc">/</span>release_39<span class="sc">/</span>gencode.v39.metadata.EntrezGene.gz<span class="sc">&gt;</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="er">&lt;</span>https<span class="sc">:</span><span class="er">//</span>ftp.ebi.ac.uk<span class="sc">/</span>pub<span class="sc">/</span>databases<span class="sc">/</span>gencode<span class="sc">/</span>Gencode_human<span class="sc">/</span>release_39<span class="sc">/</span>gencode.v39.metadata.HGNC.gz<span class="sc">&gt;</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="er">&lt;</span>https<span class="sc">:</span><span class="er">//</span>ftp.ebi.ac.uk<span class="sc">/</span>pub<span class="sc">/</span>databases<span class="sc">/</span>gencode<span class="sc">/</span>Gencode_human<span class="sc">/</span>release_39<span class="sc">/</span>gencode.v39.chr_patch_hapl_scaff.basic.annotation.gtf.gz<span class="sc">&gt;</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="er">&lt;</span>https<span class="sc">:</span><span class="er">//</span>download.rgd.mcw.edu<span class="sc">/</span>pub<span class="sc">/</span>data_release<span class="sc">/</span>RGD_ORTHOLOGS.txt<span class="sc">&gt;</span> (<span class="dv">2020-01-10</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="sc">&lt;</span>https<span class="sc">:</span><span class="er">//</span>download.rgd.mcw.edu<span class="sc">/</span>data_release<span class="sc">/</span>RAT<span class="sc">/</span>GENES_RAT.txt<span class="sc">&gt;</span> (<span class="dv">2020-11-15</span>)</span></code></pre><p></p></div>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This map was compiled from several external sources.
GENCODE metadata and annotation files were used to map between human Ensembl transcript IDs, Entrez IDs,
GENCODE IDs, and Ensembl gene IDs <a href="https://pubmed.ncbi.nlm.nih.gov/33270111/" class="external-link">(Frankish et al., 2021)</a>. RGD files were used to map between human and
rat gene symbols as well as between various rat gene identifiers <a href="https://pubmed.ncbi.nlm.nih.gov/31713623/" class="external-link">(Smith et al., 2020)</a>.</p>
<p></p><div class="sourceCode"><pre><code><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>There is one row per unique combination of \code{RAT_SYMBOL}, \code{RAT_GENE_RGD_ID}, \code{RAT_NCBI_GENE_ID}, \code{RAT_ENSEMBL_ID},</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>\code{RAT_UNIPROT_ID}, \code{HUMAN_ORTHOLOG_SYMBOL}, \code{HUMAN_ORTHOLOG_RGD}, \code{HUMAN_ORTHOLOG_NCBI_GENE_ID}, and \code{HUMAN_ORTHOLOG_ENSEMBL_ID}, </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>so some genes correspond to multiple rows <span class="cf">in</span> the table. For example, there are <span class="dv">24</span> rows corresponding to \code{RAT_SYMBOL <span class="sc">==</span> <span class="st">"Tnf"</span>} </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>because <span class="dv">3</span> \code{RAT_ENSEMBL_ID}s correspond to Tnf, and <span class="dv">8</span> \code{HUMAN_ORTHOLOG_ENSEMBL_ID}s correspond to TNF.  </span></code></pre><p></p></div>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

