<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Training summary statistics from differential analysis (DA)
that tests the effect of training on each feature within each sex.
One data frame per data type and tissue."><title>Training differential analysis results — TRAINING_DA • MotrpacRatTraining6moData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Training differential analysis results — TRAINING_DA"><meta property="og:description" content="Training summary statistics from differential analysis (DA)
that tests the effect of training on each feature within each sex.
One data frame per data type and tissue."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6moData.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Training differential analysis results</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>TRAINING_DA.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Training summary statistics from differential analysis (DA)
that tests the effect of training on each feature within each sex.
One data frame per data type and tissue.</p>
    </div>


    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with up to 47 variables:</p><dl><dt><code>assay</code></dt>
<dd><p>character, assay code used in data release. See <a href="https://rdrr.io/pkg/MotrpacBicQC/man/assay_codes.html" class="external-link">MotrpacBicQC::assay_codes</a>.</p></dd>

<dt><code>tissue</code></dt>
<dd><p>character, tissue code used in data release. See <a href="https://rdrr.io/pkg/MotrpacBicQC/man/bic_animal_tissue_code.html" class="external-link">MotrpacBicQC::bic_animal_tissue_code</a>.</p></dd>

<dt><code>feature_ID</code></dt>
<dd><p>character, MoTrPAC feature identifier</p></dd>

<dt><code>removed_samples</code></dt>
<dd><p>character, comma-separated list of vial labels excluded from differential analysis or NA</p></dd>

<dt><code>removed_samples_male</code></dt>
<dd><p>character, comma-separated list of male outliers (vial labels) removed from differential analysis</p></dd>

<dt><code>removed_samples_female</code></dt>
<dd><p>character, comma-separated list of female outliers (vial labels) removed from differential analysis</p></dd>

<dt><code>dataset</code></dt>
<dd><p>character, specific dataset</p></dd>

<dt><code>dataset_male</code></dt>
<dd><p>character, specific dataset</p></dd>

<dt><code>dataset_female</code></dt>
<dd><p>character, specific dataset</p></dd>

<dt><code>panel</code></dt>
<dd><p>character, LUMINEX panel for immunoassay analytes</p></dd>

<dt><code>full_model</code></dt>
<dd><p>character, full model used for differential analysis</p></dd>

<dt><code>full_model_male</code></dt>
<dd><p>character, full model used for differential analysis in males</p></dd>

<dt><code>full_model_female</code></dt>
<dd><p>character, full model used for differential analysis in females</p></dd>

<dt><code>reduced_model</code></dt>
<dd><p>character, effective reduced model used for differential analysis</p></dd>

<dt><code>reduced_model_male</code></dt>
<dd><p>character, effective reduced model used for differential analysis in males</p></dd>

<dt><code>reduced_model_female</code></dt>
<dd><p>character, effective reduced model used for differential analysis in females</p></dd>

<dt><code>covariates</code></dt>
<dd><p>character, comma-separated list of adjustment variables</p></dd>

<dt><code>covariates_male</code></dt>
<dd><p>character, comma-separated list of adjustment variables used for males</p></dd>

<dt><code>covariates_female</code></dt>
<dd><p>character, comma-separated list of adjustment variables used for females</p></dd>

<dt><code>fscore_male</code></dt>
<dd><p>double, F-statistic for males</p></dd>

<dt><code>fscore_female</code></dt>
<dd><p>double, F-statistic for females</p></dd>

<dt><code>lrt</code></dt>
<dd><p>double, likelihood ratio test statistic when only one sex exists</p></dd>

<dt><code>lrt_male</code></dt>
<dd><p>double, likelihood ratio test statistic for males</p></dd>

<dt><code>lrt_female</code></dt>
<dd><p>double, likelihood ratio test statistic for females</p></dd>

<dt><code>p_value_male</code></dt>
<dd><p>double, nominal p-value for males</p></dd>

<dt><code>p_value_female</code></dt>
<dd><p>character, nominal p-value for females</p></dd>

<dt><code>adj_p_value_male</code></dt>
<dd><p>double, IHW-adjusted p-value for males</p></dd>

<dt><code>adj_p_value_female</code></dt>
<dd><p>double, IHW-adjusted p-value for females</p></dd>

<dt><code>p_value</code></dt>
<dd><p>double, combined male and female nominal p-value using the sum of logs</p></dd>

<dt><code>adj_p_value</code></dt>
<dd><p>double, IHW-adjusted combined p-value; value provided by <code>selection_fdr</code> in the timewise summary statistics</p></dd>

<dt><code>tissue_abbreviation</code></dt>
<dd><p>character, tissue abbreviation, one of <a href="TISSUE_ABBREV.html">TISSUE_ABBREV</a></p></dd>

<dt><code>Chr</code></dt>
<dd><p>character, chromosome where feature is located. METHYL only.</p></dd>

<dt><code>Locus</code></dt>
<dd><p>character, name of feature. METHYL only.</p></dd>

<dt><code>EntrezID</code></dt>
<dd><p>integer, Entrez ID of corresponding gene. METHYL only.</p></dd>

<dt><code>Symbol</code></dt>
<dd><p>character, gene symbol of corresponding gene. METHYL only.</p></dd>

<dt><code>is_targeted</code></dt>
<dd><p>logical, whether metabolite is targeted. Metabolomics only.</p></dd>

<dt><code>site</code></dt>
<dd><p>character, Chemical Analysis Site (CAS) name. Metabolomics only.</p></dd>

<dt><code>metabolite</code></dt>
<dd><p>character, name of metabolite as appears in the CAS's data. Metabolomics only.</p></dd>

<dt><code>metabolite_refmet</code></dt>
<dd><p>character, RefMet name of metabolite. Metabolomics only.</p></dd>

<dt><code>groups_tested_male</code></dt>
<dd><p>character, timepoints used to perform the F-test in males.
Some tissues or assays are missing timepoints. Metabolomics only.</p></dd>

<dt><code>groups_tested_female</code></dt>
<dd><p>character, timepoints used to perform the F-test in females.
Some tissues or assays are missing timepoints. Metabolomics only.</p></dd>

<dt><code>mz</code></dt>
<dd><p>double, mass over charge. Metabolomics only.</p></dd>

<dt><code>rt</code></dt>
<dd><p>double, retention time. Metabolomics only.</p></dd>

<dt><code>neutral_mass</code></dt>
<dd><p>numeric, neutral mass. Metabolomics only.</p></dd>

<dt><code>cv</code></dt>
<dd><p>double, feature coffeicient of variation in the dataset. Metabolomics only.</p></dd>

<dt><code>meta_reg_het_p</code></dt>
<dd><p>double, heterogeneity p-value from meta-regression. A smaller p-value indicates more disagreement between platforms.
Metabolomics meta-regression only.</p></dd>

<dt><code>original_ftest_ps</code></dt>
<dd><p>character, comma-separated list of original F-test p-values. Metabolomics meta-regression only.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>While the more commonly-used timewise summary statistics are available directly in
this package (e.g., <a href="TRNSCRPT_DA.html">TRNSCRPT_DA</a>, <a href="PROTEOME_DA.html">PROTEOME_DA</a>, <a href="IMMUNO_DA.html">IMMUNO_DA</a>, <a href="METAB_DA.html">METAB_DA</a>, <a href="METAB_DA_METAREG.html">METAB_DA_METAREG</a>, <a href="TRAINING_REGULATED_FEATURES.html">TRAINING_REGULATED_FEATURES</a>, <a href="REPFDR_INPUTS.html">REPFDR_INPUTS</a>),
training summary statistics are available for download from Google Cloud Storage.
Use <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_training_da.html" class="external-link">MotrpacRatTraining6mo::load_training_da()</a></code> to download results for a given tissue and ome.
Note that the adjusted p-value from the training differential analysis is already included in the
timewise summary statistics as <code>selection_fdr</code>.</p>
<p>Training differential analysis was performed using either F-tests or likelihood
ratio tests depending on the ome. Find the functions used to generate these results listed
<a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/index.html#perform-differential-analysis" class="external-link">here</a>.
See details in the supplementary methods of the <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData#citing-motrpac-data" class="external-link">manuscript</a>.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

