<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="RNA-seq experimental and quantification QC metrics for transcriptomic (TRNSCRPT) data"><title>RNA-seq metadata and QC — TRNSCRPT_META • MotrpacRatTraining6moData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="RNA-seq metadata and QC — TRNSCRPT_META"><meta property="og:description" content="RNA-seq experimental and quantification QC metrics for transcriptomic (TRNSCRPT) data"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6moData.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>RNA-seq metadata and QC</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>TRNSCRPT_META.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>RNA-seq experimental and quantification QC metrics for transcriptomic (TRNSCRPT) data</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">TRNSCRPT_META</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame with 935 rows and 82 variables:</p><dl><dt><code>viallabel</code></dt>
<dd><p>character, sample identifier</p></dd>

<dt><code>vial_label</code></dt>
<dd><p>double, sample identifier</p></dd>

<dt><code>2D_barcode</code></dt>
<dd><p>double, sample barcode</p></dd>

<dt><code>Species</code></dt>
<dd><p>character, species</p></dd>

<dt><code>BID</code></dt>
<dd><p>integer, biospecimen ID</p></dd>

<dt><code>PID</code></dt>
<dd><p>double, participant ID, one per animal</p></dd>

<dt><code>Tissue</code></dt>
<dd><p>character, tissue description</p></dd>

<dt><code>Sample_category</code></dt>
<dd><p>character, study sample ("study") or reference standard ("ref)</p></dd>

<dt><code>GET_site</code></dt>
<dd><p>character, which Genomics, Epigenomics, and Transcriptomics (GET)
site performed the assay, "Stanford" or "MSSM" (Icahn School of Medicine at Mount Sinai)</p></dd>

<dt><code>RNA_extr_plate_ID</code></dt>
<dd><p>character, RNA extraction plate ID</p></dd>

<dt><code>RNA_extr_date</code></dt>
<dd><p>character, RNA extraction date</p></dd>

<dt><code>RNA_extr_conc</code></dt>
<dd><p>double, RNA concentration (ng/uL)</p></dd>

<dt><code>RIN</code></dt>
<dd><p>double, RNA Integrity Number</p></dd>

<dt><code>r_260_280</code></dt>
<dd><p>double, 260/280 ratio</p></dd>

<dt><code>r_260_230</code></dt>
<dd><p>double 260/230 ratio</p></dd>

<dt><code>Lib_prep_date</code></dt>
<dd><p>character, library preparation date in MM/DD/YYYY format</p></dd>

<dt><code>Lib_RNA_conc</code></dt>
<dd><p>double, RNA concentration used for library prep (ng/uL)</p></dd>

<dt><code>Lib_RNA_vol</code></dt>
<dd><p>integer, RNA volume used for library prep (uL)</p></dd>

<dt><code>Lib_robot</code></dt>
<dd><p>character, robot used for library prep</p></dd>

<dt><code>Lib_vendor</code></dt>
<dd><p>character, library prep vendor</p></dd>

<dt><code>Lib_type</code></dt>
<dd><p>character, library prep type</p></dd>

<dt><code>Lib_kit_id</code></dt>
<dd><p>character, library prep kit ID</p></dd>

<dt><code>Lib_batch_ID</code></dt>
<dd><p>character, library prep batch ID that distinguished different sample processing batches</p></dd>

<dt><code>Lib_barcode_well</code></dt>
<dd><p>character, well</p></dd>

<dt><code>Lib_index_1</code></dt>
<dd><p>character, i7 index</p></dd>

<dt><code>Lib_index_2</code></dt>
<dd><p>character, i5 index</p></dd>

<dt><code>Lib_adapter_1</code></dt>
<dd><p>character, Truseq I7 index with 16bp index</p></dd>

<dt><code>Lib_adapter_2</code></dt>
<dd><p>character, Truseq I5 index with 8bp index</p></dd>

<dt><code>Lib_UMI_cycle_num</code></dt>
<dd><p>integer, number of bases of UMI</p></dd>

<dt><code>Lib_adapter_size</code></dt>
<dd><p>integer, total size of the two adapters</p></dd>

<dt><code>Lib_frag_size</code></dt>
<dd><p>integer, average library fragment size (bp)</p></dd>

<dt><code>Lib_DNA_conc</code></dt>
<dd><p>double, DNA concentration of original stock of the library (ng/uL)</p></dd>

<dt><code>Lib_molarity</code></dt>
<dd><p>double, library molarity (nM)</p></dd>

<dt><code>Seq_platform</code></dt>
<dd><p>character, sequencing platform</p></dd>

<dt><code>Seq_date</code></dt>
<dd><p>integer, sequencing date, YYMMDD format</p></dd>

<dt><code>Seq_machine_ID</code></dt>
<dd><p>character, serial number of the sequencer</p></dd>

<dt><code>Seq_flowcell_ID</code></dt>
<dd><p>character, flow cell ID</p></dd>

<dt><code>Seq_flowcell_run</code></dt>
<dd><p>integer, flow cell run</p></dd>

<dt><code>Seq_flowcell_lane</code></dt>
<dd><p>character, flow cell lane</p></dd>

<dt><code>Seq_flowcell_type</code></dt>
<dd><p>character, flow cell type, e.g., S4</p></dd>

<dt><code>Seq_length</code></dt>
<dd><p>integer, read length</p></dd>

<dt><code>Seq_end_type</code></dt>
<dd><p>integer, 1=single-end, 2=paired-end</p></dd>

<dt><code>Phase</code></dt>
<dd><p>character, study phase, "PASS1B-06"</p></dd>

<dt><code>Seq_batch</code></dt>
<dd><p>character, unique identifier for sequencing batch</p></dd>

<dt><code>reads_raw</code></dt>
<dd><p>double, number of read pairs in the raw FASTQ</p></dd>

<dt><code>pct_adapter_detected</code></dt>
<dd><p>double, percent of reads with adapter detected</p></dd>

<dt><code>pct_trimmed</code></dt>
<dd><p>double, percent of reads that were trimmed</p></dd>

<dt><code>pct_trimmed_bases</code></dt>
<dd><p>double, percent of bases that were trimmed</p></dd>

<dt><code>reads</code></dt>
<dd><p>double, number of read pairs in the trimmed FASTQ files</p></dd>

<dt><code>pct_GC</code></dt>
<dd><p>double, percent GC content in trimmed FASTQ files</p></dd>

<dt><code>pct_dup_sequence</code></dt>
<dd><p>double, percent of duplicated sequences in trimmed FASTQ files</p></dd>

<dt><code>pct_rRNA</code></dt>
<dd><p>double, percent of rRNA reads in trimmed FASTQ files</p></dd>

<dt><code>pct_globin</code></dt>
<dd><p>double, percent of globin reads in trimmed FASTQ files</p></dd>

<dt><code>pct_phix</code></dt>
<dd><p>double, percent of phix reads in trimmed FASTQ files</p></dd>

<dt><code>pct_picard_dup</code></dt>
<dd><p>double, PCR duplication assessed by Picard’s tool MarkDuplicate</p></dd>

<dt><code>pct_umi_dup</code></dt>
<dd><p>double, PCR duplication rate assessed using UMIs (Unique Molecular Identifiers)</p></dd>

<dt><code>avg_input_read_length</code></dt>
<dd><p>double, average input read length</p></dd>

<dt><code>uniquely_mapped</code></dt>
<dd><p>double, number of uniquely mapped reads</p></dd>

<dt><code>pct_uniquely_mapped</code></dt>
<dd><p>double, percent of uniquely mapped reads</p></dd>

<dt><code>avg_mapped_read_length</code></dt>
<dd><p>double, average input mapped length</p></dd>

<dt><code>num_splices</code></dt>
<dd><p>double, number of splices</p></dd>

<dt><code>num_annotated_splices</code></dt>
<dd><p>double, number of annotated splices</p></dd>

<dt><code>num_GTAG_splices</code></dt>
<dd><p>double, number of GT/AG and CT/AC splices</p></dd>

<dt><code>num_GCAG_splices</code></dt>
<dd><p>double, number of GC/AG and CT/GC splices</p></dd>

<dt><code>num_ATAC_splices</code></dt>
<dd><p>double, number of AT/AC and GT/TA splices</p></dd>

<dt><code>num_noncanonical_splices</code></dt>
<dd><p>double, number of non-canonical splices</p></dd>

<dt><code>pct_multimapped</code></dt>
<dd><p>double, percent of reads that multimapped</p></dd>

<dt><code>pct_multimapped_toomany</code></dt>
<dd><p>double, percent of reads that multimapped too many times</p></dd>

<dt><code>pct_unmapped_mismatches</code></dt>
<dd><p>double, percent of unmapped reads due to mismatches</p></dd>

<dt><code>pct_unmapped_tooshort</code></dt>
<dd><p>double, percent of unmapped reads due to shortness</p></dd>

<dt><code>pct_unmapped_other</code></dt>
<dd><p>double, percent of unmapped reads for other reason</p></dd>

<dt><code>pct_chimeric</code></dt>
<dd><p>double, percent chimeric reads</p></dd>

<dt><code>pct_chrX</code></dt>
<dd><p>double, percent of reads mapped to chromosome X</p></dd>

<dt><code>pct_chrY</code></dt>
<dd><p>double, percent of reads mapped to chromosome Y</p></dd>

<dt><code>pct_chrM</code></dt>
<dd><p>double, percent of reads mapped to the mitochondrial genome</p></dd>

<dt><code>pct_chrAuto</code></dt>
<dd><p>double, percent of reads mapped to autosomal chromosomes</p></dd>

<dt><code>pct_contig</code></dt>
<dd><p>double, percent of reads mapped to contigs</p></dd>

<dt><code>pct_coding</code></dt>
<dd><p>double, percent of bases mapped to coding</p></dd>

<dt><code>pct_utr</code></dt>
<dd><p>double, percent of bases mapped to untranslated region</p></dd>

<dt><code>pct_intronic</code></dt>
<dd><p>double, percent of bases mapped to introns</p></dd>

<dt><code>pct_intergenic</code></dt>
<dd><p>double, percent of bases mapped to intergenic</p></dd>

<dt><code>pct_mrna</code></dt>
<dd><p>double, percent of bases mapped to mRNA</p></dd>

<dt><code>median_5_3_bias</code></dt>
<dd><p>double, median 5' to 3' bias</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="source">Source<a class="anchor" aria-label="anchor" href="#source"></a></h2>
    <p><code>pass1b-06/results/transcriptomics/qa-qc/motrpac_pass1b-06_transcript-rna-seq_qa-qc-metrics.csv</code></p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

