---
title: "Download sample-level data"
author: "Nicole Gay"
date: "12/9/2021"
output: html_document
---

```{r setup, include=FALSE}
## WARNING: this code is not self-contained. it requires access to the motrpac-mawg GitHub and gs://mawg-data GCS 
library(data.table)
library(MotrpacBicQC)
library(dplyr)
library(testit)
library(usethis)
library(sinew)
library(devtools)
knitr::opts_chunk$set(echo = TRUE)
mawgdir = 'nicolerg/MOTRPAC/MAWG_DATA'
gitdir = "nicolerg/src/MOTRPAC/"
scratch = 'nicolerg/tmp'
datadir = 'motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data'
gsutil = 'gsutil'
source(sprintf("%s/motrpac-mawg/pass1b-06/tools/get_fx.R", gitdir))
source(sprintf("%s/motrpac-mawg/pass1b-06/integrative/clustering/cluster_viz_fx.R", gitdir))
#load_all("../../R")
```

```{r load data}
load("../../data/REPEATED_FEATURES.rda")
load("../../data/TRAINING_REGULATED_FEATURES.rda")
differential_features = unique(TRAINING_REGULATED_FEATURES$feature)
repeated_feature_map = as.data.table(REPEATED_FEATURES)
# # feature_to_universe and 5% FDR tables
# # system(sprintf("gsutil -m cp -r gs://mawg-data/pass1b-06/merged %s", datadir))
# load(sprintf("%s/merged/zs_smoothed-0.05fdr-0logfc-cluster-input_20211115.RData", mawgdir))
# load(sprintf("%s/merged/master_feature_to_gene_20211116.RData", mawgdir))
# 
# differential_features = rownames(zs_smoothed)
# master_feature_to_gene = NULL # don't need this 
# head(repeated_feature_map)
# read in pheno data
pheno = dl_format_pheno(scratch, gsutil, parallel=T)
```

```{r fx}
combine_normalized_data = function(gsutil_path, 
                                   pattern, 
                                   ome_abbreviation, 
                                   datadir, 
                                   diff_feat = differential_features,
                                   gsutil_bin = 'gsutil', 
                                   pheno = NULL, 
                                   features_to_keep = NULL, 
                                   tissue_index=3, 
                                   skip=1, 
                                   n_rows=Inf,
                                   file_list=NULL, 
                                   viallabel=F, 
                                   add_filename_as_column=F,
                                   return_list=T,
                                   colname_type="viallabel",
                                   rename_features=F){
  norm_list = list()
  
  if(is.null(file_list)){
    # list normalized data
    file_list = system(sprintf("%s ls %s/*%s*", gsutil_bin, gsutil_path, pattern), intern=T)
    if(length(file_list)==0){
      stop(sprintf("No files found for %s/*%s*", gsutil_path, pattern))
    }
    # check if files already exist
    file_names = paste(sprintf("%s/%s", datadir, ome_abbreviation), basename(file_list), sep="/")
    if(!all(file.exists(file_names))){
      # download those files 
      system(sprintf("mkdir -p %s/%s", datadir, ome_abbreviation))
      system(sprintf("%s -m cp %s/*%s* %s/%s", gsutil_bin, gsutil_path, pattern, datadir, ome_abbreviation))
    }
    file_list = list.files(path=sprintf("%s/%s", datadir, ome_abbreviation), pattern=pattern,  full.names = T)
  }
  # read in and format 
  for(f in file_list){
    if(ome_abbreviation=='ATAC' & grepl("_mssm_", f)) next
    print(basename(f))
    if(colname_type == "pid"){
      curr_norm = fread(f, sep='\t', header=T, skip = skip, nrows=n_rows) # use pid as colname 
      if(colnames(curr_norm)[1]!='pid'){
        warning(sprintf("pid is not where expected in the header for %s. Renaming column.",f))
        colnames(curr_norm)[1] = 'pid'
      }else{
        # remove bid
        curr_norm = curr_norm[2:nrow(curr_norm)]
      }
    }else if(colname_type == "viallabel"){
      curr_norm = fread(f, sep='\t', header=T, skip = 0, nrows=n_rows) 
      colname = "viallabel"
      if(colnames(curr_norm)[1]!='viallabel'){
        warning(sprintf("viallabel is not where expected in the header for %s.",f))
        print(head(curr_norm))
        if(colnames(curr_norm)[1] == "viallabel-Me/Un"){
          colname = "viallabel-Me/Un"
        }else if(colnames(curr_norm)[1] == "chrom"){
          if(ome_abbreviation == "ATAC"){
            # make "feature_ID"
            curr_norm = data.table(cbind(
              feature_ID = sprintf("%s:%s-%s",
                                   curr_norm[,chrom],
                                   curr_norm[,start],
                                   curr_norm[,end]),
              curr_norm[,4:ncol(curr_norm), with=F]
            ))
          }else{
            stop()
          }
        }else{
          stop()
        }
      }
      # remove pid, bid
      if(curr_norm[1,get(colname)] == "pid" & curr_norm[2,get(colname)] == "bid"){
        curr_norm = curr_norm[3:nrow(curr_norm)]
      }
    }
    tissue_code = unname(unlist(strsplit(basename(f), '_')))[tissue_index]
    if(!tissue_code %in% names(tissue_abbr)){
      stop(sprintf("Tissue code not recognized for %s: %s", f, tissue_code))
    }
    tissue_abbrv = tissue_abbr[[tissue_code]]
    
    if(rename_features){
      # fix feature id
      feature_names = sprintf("%s;%s;%s", ome_abbreviation, tissue_abbrv, curr_norm[,get(colname_type)])
      curr_norm[,(colname_type) := NULL]
      curr_norm = cbind(data.table(feature=feature_names), curr_norm)
    }else{
      print(colnames(curr_norm)[1])
      print(colname_type)
      setnames(curr_norm, colnames(curr_norm)[1], "feature_ID")
    }
    
    # replace viallabel with pid
    if(viallabel){
      if(is.null(pheno)) stop("pheno needs to be supplied if viallabel=T")
      # replace viallabel with pid
      print(colnames(curr_norm) %in% as.character(pheno[,viallabel]))
      pheno_sub = unique(pheno[,.(viallabel, pid)])
      colnames(curr_norm)[colnames(curr_norm) %in% as.character(pheno[,viallabel])] = 
        as.character(pheno_sub[match(colnames(curr_norm)[colnames(curr_norm) %in% as.character(pheno[,viallabel])], as.character(viallabel)), pid])
    }
    
    if(add_filename_as_column){
      curr_norm[,origin_file := basename(f)]
    }
    
    # add cols: feature, feature_ID, tissue (tissue abbreviation), assay (assay/ome abbreviation)
    curr_norm = data.table(cbind(feature = sprintf("%s;%s;%s", ome_abbreviation, tissue_abbrv, curr_norm[,feature_ID]),
                                 feature_ID = curr_norm[,feature_ID],
                                 tissue = tissue_abbrv,
                                 assay = ome_abbreviation,
                                 curr_norm[,2:ncol(curr_norm), with=F]))
    
    if(!is.null(features_to_keep)){
      # filter down to these features
      print(dim(curr_norm))
      print(length(features_to_keep))
      curr_norm = curr_norm[feature %in% features_to_keep]
      print(dim(curr_norm))
    }
    
    # remove feature for non-differential features 
    curr_norm[!feature %in% diff_feat, feature := NA]
    
    norm_list[[tissue_abbrv]] = curr_norm
    
  }
  
  if(return_list){
    return(norm_list)
  }
  
  norm = rbindlist(norm_list, fill=T)
  
  if(!is.null(pheno)){
    # check if pids are valid 
    if(colname_type == "pid"){
      if(!all(colnames(norm)[2:ncol(norm)] %in% as.character(pheno[,pid]))){
        warning("Some PIDs not found")
      }
    }
    if(colname_type == "viallabel"){
      if(!all(colnames(norm)[2:ncol(norm)] %in% as.character(pheno[,pid]))){
        warning("Some PIDs not found")
      }
    }
  }
  
  return(norm)
}
viallabel_to_pid = function(viallabels){
  pheno = as.data.table(MotrpacRatTraining6moData::PHENO)
  pheno = unique(pheno[,.(viallabel, pid)])
  pheno = pheno[viallabel %in% as.character(viallabels)]
  vl_to_pid = pheno[,pid]
  names(vl_to_pid) = pheno[,viallabel]
  return(vl_to_pid)
}
```

## Normalized data

### Load normalized data for TRNSCRPT, proteomics 
```{r read in for each ome}
norm_data_list = list()
# rna
norm_data_list[["TRNSCRPT"]] = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/transcriptomics/transcript-rna-seq/normalized-data",
                                                       "normalized-log-cpm", "TRNSCRPT", datadir, pheno=pheno)
# prot-pr
prot = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-pr/normalized-data", 
                               "med-mad-normalized-logratio.txt", "PROT", datadir, pheno=pheno)
norm_data_list[['PROT']] = prot
# prot-ac
ac = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ac/normalized-data", 
                             "med-mad-normalized-logratio.txt", "ACETYL", datadir, pheno=pheno)
norm_data_list[['ACETYL']] = ac
# prot-ub
ub = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ub/normalized-data", 
                             "med-mad-normalized-protein-corrected-logratio.txt", "UBIQ", datadir, pheno=pheno)
norm_data_list[['UBIQ']] = ub  
# prot-ph
ph = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ph/normalized-data", 
                             "med-mad-normalized-logratio.txt", "PHOSPHO", datadir, pheno=pheno)
norm_data_list[['PHOSPHO']] = ph
```

### Load ATAC, RRBS normalized data only for training-regulated features 
```{r}
# atac
atac = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-atac-seq/normalized-data", 
                               "quant-norm", "ATAC", datadir, pheno=pheno, 
                               features_to_keep = differential_features[grepl("ATAC", differential_features)], tissue_index = 4)
length(differential_features[grepl("ATAC", differential_features)])
sum(unlist(lapply(atac, nrow)))
norm_data_list[['ATAC']] = atac
# rrbs
rrbs = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", 
                               "normalized-log-M-window.txt", "METHYL", datadir, pheno=pheno, 
                               features_to_keep = differential_features[grepl("METHYL", differential_features)])
length(differential_features[grepl("METHYL", differential_features)])
sum(unlist(lapply(rrbs, nrow)))
norm_data_list[['METHYL']] = rrbs
```

```{r save list}
save(norm_data_list, file=sprintf("%s/norm_data_list.RData", datadir))
```

### Save one .rda per tissue per ome to R package
```{r save one per dataset}
# save one table per ome and tissue
load(file=sprintf("%s/norm_data_list.RData", datadir))
for(ome in names(norm_data_list)){
  
  for(tissue in names(norm_data_list[[ome]])){
    
    norm_data = norm_data_list[[ome]][[tissue]]
    
    # make sure values are numeric
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
    
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    if(ome %in% c("ATAC","METHYL")){
      new_name = sprintf("%s_%s_NORM_DATA_05FDR",ome,gsub("-","",tissue))
    }else{
      new_name = sprintf("%s_%s_NORM_DATA",ome,gsub("-","",tissue))
    }
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))
    
    # Save .rda - this keeps giving me an error early in the loop
    do.call("use_data", list(as.name(new_name), overwrite = TRUE))
  }
}
```

### Handle IMMUNO normalized data separately 

#### Nested IMMUNO data for DA and IMMUNO sample metadata 

Protein extractions from the same viallabel were used in multiple panels. 
Furthermore, not all assays were performed on every viallabel. 
CHEX* variables are also well-specific measurements, where CHEX4 is used as a covariate in differential analysis. 
```{r immuno}
metadata = dl_read_gcp("gs://mawg-data/pass1b-06/immunoassay/data/release/sample_metadata_20210803.csv", sep=',')
metadata[,group := ifelse(intervention == 'control', 'control', sacrifice_time)]
metadata[,c('intervention','sacrifice_time') := NULL]
# version of the data used for differential analysis 
flist = system("gsutil ls gs://mawg-data/pass1b-06/immunoassay/data/release/pass1b-06*_mfi-log2-filt-imputed-na-outliers.txt", intern=T)
# add CHEX to metadata
new_meta_list = list()
imputed_data = list()
for(f in flist){
  splits = unname(unlist(strsplit(basename(f), '_')))
  panel = splits[4]
  if(!panel %in% metadata[,panel_name] & panel %in% c("ADIPONECTIN","SERPIN-E")){
    meta_panel = "rat-adipokine"
  }else{
    meta_panel = panel
  }
  tissue = splits[2]
  tissue_abbrev = MotrpacBicQC::tissue_abbr[[tissue]]
  print(sprintf("%s %s", panel, tissue))
  if(!panel %in% names(imputed_data)){
    imputed_data[[panel]] = list()
  }
  if(!tissue %in% names(imputed_data[[panel]])){
    imputed_data[[panel]][[tissue_abbrev]] = list()
  }
  # add CHEX to metadata
  imputed = dl_read_gcp(f, sep='\t')
  imputed[,vial_label := as.character(vial_label)]
  new_meta_list[[sprintf("%s;%s", panel, tissue)]] = merge(metadata[tissue_code==tissue & panel_name==meta_panel], 
                                                           imputed[,.(vial_label, CHEX1, CHEX2, CHEX3, CHEX4)],
                                                           by="vial_label")
  # remove CHEX from imputed data
  imputed[,c("CHEX1","CHEX2","CHEX3","CHEX4") := NULL]
  
  setnames(imputed, "vial_label", "viallabel")
  imputed = as.data.frame(imputed)
  imputed_data[[panel]][[tissue_abbrev]] = imputed
}
new_meta = rbindlist(new_meta_list)
# add ref stds
metadata[!vial_label %in% new_meta[,vial_label]]
new_meta = rbindlist(list(new_meta, metadata[!vial_label %in% new_meta[,vial_label]]), fill=T)
setnames(new_meta, "vial_label", "viallabel")
new_meta[new_meta==""] = NA
# rename CHEX columns - these are already log2
setnames(new_meta,
         c("CHEX1","CHEX2","CHEX3","CHEX4"),
         c("log2_CHEX1","log2_CHEX2","log2_CHEX3","log2_CHEX4"))
# save data
IMMUNO_META = as.data.frame(new_meta)
usethis::use_data(IMMUNO_META, overwrite=T)
sinew::makeOxygen(IMMUNO_META)
# write metadata to file 
write.table(IMMUNO_META, file=sprintf("%s/sample_metadata_20220727.csv",scratch), sep=",", col.names=T, row.names=F, quote=F)
system(sprintf("gsutil cp %s gs://mawg-data/pass1b-06/immunoassay/data/release/",
               sprintf("%s/sample_metadata_20220727.csv",scratch)))
# save the data in list format 
IMMUNO_NORM_DATA_NESTED = imputed_data
usethis::use_data(IMMUNO_NORM_DATA_NESTED, overwrite=T)
```

#### Single normalized data table

Because there are multiple viallabels per animal per tissue for IMMUNO data,
make a single normalized data table where column names are PID instead of viallabel. 

```{r format IMMUNO for viz}
load("../../data/IMMUNO_NORM_DATA_NESTED.rda")
load("../../data/REPEATED_FEATURES.rda")
rep = data.table(REPEATED_FEATURES)
names(IMMUNO_NORM_DATA_NESTED)
data_list = list()
for(panel in names(IMMUNO_NORM_DATA_NESTED)){
  for(tissue in names(IMMUNO_NORM_DATA_NESTED[[panel]])){
    df = IMMUNO_NORM_DATA_NESTED[[panel]][[tissue]]
    # remove refstd
    df = df[!grepl("^8", df$viallabel),]
    # transpose
    rownames(df) = df$viallabel
    df$viallabel = NULL
    df_t = as.data.frame(t(df))
    feature_ID = rownames(df_t)
    # fix repeated features
    features = sprintf("IMMUNO;%s;%s", tissue, feature_ID)
    if(any(features %in% rep[,feature])){
      new_features = sprintf("IMMUNO;%s;%s:%s", tissue, panel, feature_ID)
      features[new_features %in% rep[,new_feature]] = new_features[new_features %in% rep[,new_feature]]
    }
    # replace colname with pid 
    colnames(df_t) = viallabel_to_pid(colnames(df_t))
    # make data.table
    dt = as.data.table(cbind(feature=features, 
                             feature_ID=feature_ID,
                             tissue=tissue, 
                             assay="IMMUNO",
                             dataset=panel,
                             df_t))
    label=sprintf("%s;%s", tissue, panel)
    writeLines(label)
    data_list[[label]] = dt
  }
}
immuno_data = rbindlist(data_list, fill=T)
# make sure rownames are unique
length(immuno_data[,feature]) == length(unique(immuno_data[,feature]))
# check that repeated features match
all(immuno_data[grepl(":",feature),feature] %in% rep[,new_feature])
# remove feature for non-differential features 
immuno_data[!feature %in% differential_features, feature := NA]
# save 
IMMUNO_NORM_DATA_FLAT = as.data.frame(immuno_data)
usethis::use_data(IMMUNO_NORM_DATA_FLAT, overwrite=T)
```

## TRNSCRPT raw counts
```{r}
data_dir = "motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/results/transcriptomics"
for(tissue_code in names(TISSUE_CODE_TO_ABBREV)){
  file = sprintf("%s/%s/transcript-rna-seq/motrpac_pass1b-06_%s_transcript-rna-seq_rsem-genes-count.txt", data_dir, tissue_code, tissue_code)
  if(file.exists(file)){
    counts = fread(file, sep="\t", header=T)
    
    # add cols: feature, feature_ID, tissue (tissue abbreviation), assay (assay/ome abbreviation)
    counts = data.table(cbind(feature = sprintf("TRNSCRPT;%s;%s", TISSUE_CODE_TO_ABBREV[[tissue_code]], counts[,gene_id]),
                                 feature_ID = counts[,gene_id],
                                 tissue = TISSUE_CODE_TO_ABBREV[[tissue_code]],
                                 assay = "TRNSCRPT",
                                 counts[,2:ncol(counts), with=F]))
    
    # remove feature for non-DE features
    counts[!feature %in% differential_features, feature := NA]
    
    # remove ref stds
    cols_to_remove = colnames(counts)[grepl("^8", colnames(counts))]
    if(length(cols_to_remove)>0){
      counts[,c(cols_to_remove) := NULL]
    }
    
    # make sure values are integer
    numeric_cols = colnames(counts)[5:ncol(counts)]
    counts[,(numeric_cols) := lapply(.SD, as.integer), .SDcols = numeric_cols]
    
    # change to df for ease of use 
    df = as.data.frame(counts)
    
    # change name
    new_name = sprintf("TRNSCRPT_%s_RAW_COUNTS", gsub("-","",TISSUE_CODE_TO_ABBREV[[tissue_code]]))
    writeLines(new_name)
    assign(new_name, df)
    # save .rda
    do.call("use_data", list(as.name(new_name), overwrite = TRUE))
  }else{
    print(sprintf("file %s DNE", file))
  }
}
```

## Save full eipgenetic data sets to GCS

### ATAC 

#### Normalized data 
```{r atac norm counts}
# atac
atac = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-atac-seq/normalized-data", 
  "quant-norm", "ATAC", datadir, pheno=pheno, tissue_index = 4)
# write one file per tissue
ome="ATAC"
for(tissue in names(atac)){
  
    norm_data = atac[[tissue]]
    
    # make sure values are numeric
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
    
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    new_name = sprintf("%s_%s_NORM_DATA",ome,gsub("-","",tissue))
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    outfile = sprintf("%s/ATAC/%s.rda", datadir, new_name)
    print(outfile)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))
    # save rda
    do.call("save", list(as.name(new_name), 
                         file = outfile, 
                         compress = "bzip2", compression_level = 9))
}
# save space
atac=NULL
```

#### Raw counts 
```{r ATAC raw counts}
# atac
atac = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/results/epigenomics/*/epigen-atac-seq",
  "epigen-atac-seq_counts", "ATAC", datadir, pheno=pheno, tissue_index = 3)
meta = data.table(ATAC_META)
meta = meta[GET_site == "Stanford"]
# write one file per tissue
ome="ATAC"
for(tissue in names(atac)){
  
    norm_data = atac[[tissue]]
    
    # remove refstd
    to_remove = colnames(norm_data)[grepl("^8", colnames(norm_data))]
    if(length(to_remove)>0){
      norm_data[,c(to_remove) := NULL]
    }
    
    # make sure values are integer
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.integer), .SDcols = numeric_cols]
    
    # remove Sinai columns
    to_remove = numeric_cols[!numeric_cols %in% meta[,viallabel]]
    if(length(to_remove)>0){
      norm_data[,c(to_remove) := NULL]
    }
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    new_name = sprintf("%s_%s_RAW_COUNTS",ome,gsub("-","",tissue))
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    outfile = sprintf("%s/ATAC/%s.rda", datadir, new_name)
    print(outfile)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))
    # save rda
    do.call("save", list(as.name(new_name), 
                         file = outfile, 
                         compress = "bzip2", compression_level = 9))
}
# save space
atac=NULL
# copy to gs://motrpac-rat-training-6mo-extdata
```

### METHYL 

#### Normalized data
```{r}
# rrbs
rrbs = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", 
  "epigen-rrbs_normalized-log-M-window", "METHYL", datadir, pheno=pheno, tissue_index = 3)
# write one file per tissue
ome="METHYL"
for(tissue in names(rrbs)){
  
    norm_data = rrbs[[tissue]]
    
    # remove refstd
    to_remove = colnames(norm_data)[grepl("^8", colnames(norm_data))]
    if(length(to_remove)>0){
      norm_data[,c(to_remove) := NULL]
    }
    
    # remove outlier row
    norm_data = norm_data[feature_ID != "is_outlier"]
    
    # make sure values are numeric 
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
    
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    new_name = sprintf("%s_%s_NORM_DATA",ome,gsub("-","",tissue))
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    outfile = sprintf("%s/METHYL/%s.rda", datadir, new_name)
    print(outfile)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))
    # save rda
    do.call("save", list(as.name(new_name), 
                         file = outfile, 
                         compress = "bzip2", compression_level = 9))
}
# save space
rrbs=NULL
# copy to gs://motrpac-rat-training-6mo-extdata
```

#### Raw counts
```{r}
# rrbs
rrbs = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", 
  "epigen-rrbs_counts-window.txt", "METHYL", datadir, pheno=pheno, tissue_index = 3)
# write one file per tissue
ome="METHYL"
for(tissue in names(rrbs)){
  
    norm_data = rrbs[[tissue]]
    
    # remove refstd
    to_remove = colnames(norm_data)[grepl("^8", colnames(norm_data))]
    if(length(to_remove)>0){
      norm_data[,c(to_remove) := NULL]
    }
    
    # remove outlier row
    norm_data = norm_data[feature_ID != "is_outlier"]
    
    # make sure values are numeric 
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.integer), .SDcols = numeric_cols]
    
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    new_name = sprintf("%s_%s_RAW_COUNTS",ome,gsub("-","",tissue))
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    outfile = sprintf("%s/METHYL/%s.rda", datadir, new_name)
    print(outfile)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))
    # save rda
    do.call("save", list(as.name(new_name), 
                         file = outfile, 
                         compress = "bzip2", compression_level = 9))
}
# save space
rrbs=NULL
# copy to gs://motrpac-rat-training-6mo-extdata
```
