---
title: "Download sample-level data"
author: "Nicole Gay"
date: "12/9/2021"
output: html_document
---

```{r setup, include=FALSE}
## WARNING: this code is not self-contained. it requires access to the motrpac-mawg GitHub and gs://mawg-data GCP 

library(data.table)
library(MotrpacBicQC)
library(MotrpacRatTraining6moData)
library(dplyr)
library(testit)
library(usethis)
library(sinew)
library(devtools)
knitr::opts_chunk$set(echo = TRUE)

mawgdir = '/oak/stanford/groups/smontgom/nicolerg/MOTRPAC/MAWG_DATA'
gitdir = "/oak/stanford/groups/smontgom/nicolerg/src/MOTRPAC/"
scratch='/oak/stanford/groups/smontgom/nicolerg/tmp'
datadir = '/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data'
gsutil='~/google-cloud-sdk/bin/gsutil'

source(sprintf("%s/motrpac-mawg/pass1b-06/tools/get_fx.R", gitdir))
source(sprintf("%s/motrpac-mawg/pass1b-06/integrative/clustering/cluster_viz_fx.R", gitdir))
```

```{r load data}
# # feature_to_universe and 5% FDR tables 
# # system(sprintf("gsutil -m cp -r gs://mawg-data/pass1b-06/merged %s", datadir))
# load(sprintf("%s/merged/zs_smoothed-0.05fdr-0logfc-cluster-input_20211115.RData", mawgdir))
# load(sprintf("%s/merged/master_feature_to_gene_20211116.RData", mawgdir))
# 
# differential_features = rownames(zs_smoothed)
# master_feature_to_gene = NULL # don't need this 
# head(repeated_feature_map)

# read in pheno data
pheno = dl_format_pheno(scratch, gsutil, parallel=T)
```

```{r fx}
combine_normalized_data = function(gsutil_path, 
                                   pattern, 
                                   ome_abbreviation, 
                                   datadir, 
                                   gsutil_bin = '~/google-cloud-sdk/bin/gsutil', 
                                   pheno = NULL, 
                                   features_to_keep = NULL, 
                                   tissue_index=3, 
                                   skip=1, 
                                   n_rows=Inf,
                                   file_list=NULL, 
                                   viallabel=F, 
                                   add_filename_as_column=F,
                                   return_list=T,
                                   colname_type="viallabel",
                                   rename_features=F){
  norm_list = list()
  
  if(is.null(file_list)){
    # list normalized data
    file_list = system(sprintf("%s ls %s/*%s*", gsutil_bin, gsutil_path, pattern), intern=T)
    if(length(file_list)==0){
      stop(sprintf("No files found for %s/*%s*", gsutil_path, pattern))
    }
    # check if files already exist
    file_names = paste(sprintf("%s/%s", datadir, ome_abbreviation), basename(file_list), sep="/")
    if(!all(file.exists(file_names))){
      # download those files 
      system(sprintf("mkdir -p %s/%s", datadir, ome_abbreviation))
      system(sprintf("%s -m cp %s/*%s* %s/%s", gsutil_bin, gsutil_path, pattern, datadir, ome_abbreviation))
    }
    file_list = list.files(path=sprintf("%s/%s", datadir, ome_abbreviation), pattern=pattern,  full.names = T)
  }

  # read in and format 
  for(f in file_list){
    if(ome_abbreviation=='ATAC' & grepl("_mssm_", f)) next
    print(basename(f))
    if(colname_type == "pid"){
      curr_norm = fread(f, sep='\t', header=T, skip = skip, nrows=n_rows) # use pid as colname 
      if(colnames(curr_norm)[1]!='pid'){
        warning(sprintf("pid is not where expected in the header for %s. Renaming column.",f))
        colnames(curr_norm)[1] = 'pid'
      }else{
        # remove bid
        curr_norm = curr_norm[2:nrow(curr_norm)]
      }
    }else if(colname_type == "viallabel"){
      curr_norm = fread(f, sep='\t', header=T, skip = 0, nrows=n_rows) 
      colname = "viallabel"
      if(colnames(curr_norm)[1]!='viallabel'){
        warning(sprintf("viallabel is not where expected in the header for %s.",f))
        print(head(curr_norm))
        if(!colnames(curr_norm)[1] == "viallabel-Me/Un"){
          stop()
        }else{
          colname = "viallabel-Me/Un"
        }
      }
      # remove pid, bid
      assert(curr_norm[1,get(colname)] == "pid")
      assert(curr_norm[2,get(colname)] == "bid")
      curr_norm = curr_norm[3:nrow(curr_norm)]
    }

    tissue_code = unname(unlist(strsplit(basename(f), '_')))[tissue_index]
    if(!tissue_code %in% names(tissue_abbr)){
      stop(sprintf("Tissue code not recognized for %s: %s", f, tissue_code))
    }
    tissue_abbrv = tissue_abbr[[tissue_code]]
    
    if(rename_features){
      # fix feature id
      feature_names = sprintf("%s;%s;%s", ome_abbreviation, tissue_abbrv, curr_norm[,get(colname_type)])
      curr_norm[,(colname_type) := NULL]
      curr_norm = cbind(data.table(feature=feature_names), curr_norm)
    }else{
      print(colnames(curr_norm)[1])
      print(colname_type)
      setnames(curr_norm, colnames(curr_norm)[1], "feature_ID")
    }
    
    if(!is.null(features_to_keep)){
      # filter down to these features
      print(dim(curr_norm))
      print(length(features_to_keep))
      curr_norm = curr_norm[feature %in% features_to_keep]
      print(dim(curr_norm))
    }
    
    # replace viallabel with pid
    if(viallabel){
      if(is.null(pheno)) stop("pheno needs to be supplied if viallabel=T")
      # replace viallabel with pid
      print(colnames(curr_norm) %in% as.character(pheno[,viallabel]))
      pheno_sub = unique(pheno[,.(viallabel, pid)])
      colnames(curr_norm)[colnames(curr_norm) %in% as.character(pheno[,viallabel])] = 
        as.character(pheno_sub[match(colnames(curr_norm)[colnames(curr_norm) %in% as.character(pheno[,viallabel])], as.character(viallabel)), pid])
    }
    
    if(add_filename_as_column){
      curr_norm[,origin_file := basename(f)]
    }
    
    head(curr_norm)
    norm_list[[tissue_abbrv]] = curr_norm
  }
  
  if(return_list){
    return(norm_list)
  }
  
  norm = rbindlist(norm_list, fill=T)
  
  if(!is.null(pheno)){
    # check if pids are valid 
    if(colname_type == "pid"){
      if(!all(colnames(norm)[2:ncol(norm)] %in% as.character(pheno[,pid]))){
        warning("Some PIDs not found")
      }
    }
    if(colname_type == "viallabel"){
      if(!all(colnames(norm)[2:ncol(norm)] %in% as.character(pheno[,pid]))){
        warning("Some PIDs not found")
      }
    }
  }
  
  return(norm)
}

viallabel_to_pid = function(viallabels){
  pheno = as.data.table(MotrpacRatTraining6moData::PHENO)
  pheno = unique(pheno[,.(viallabel, pid)])
  pheno = pheno[viallabel %in% as.character(viallabels)]
  vl_to_pid = pheno[,pid]
  names(vl_to_pid) = pheno[,viallabel]
  return(vl_to_pid)
}
```

```{r read in for each ome}
norm_data_list = list()

# rna
norm_data_list[["TRNSCRPT"]] = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/transcriptomics/transcript-rna-seq/normalized-data",
                                                       "normalized-log-cpm", "TRNSCRPT", datadir, pheno=pheno)

# prot-pr
prot = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-pr/normalized-data", 
                               "med-mad-normalized-logratio.txt", "PROT", datadir, pheno=pheno)
norm_data_list[['PROT']] = prot

# prot-ac
ac = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ac/normalized-data", 
                             "med-mad-normalized-logratio.txt", "ACETYL", datadir, pheno=pheno)
norm_data_list[['ACETYL']] = ac

# prot-ub
ub = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ub/normalized-data", 
                             "med-mad-normalized-protein-corrected-logratio.txt", "UBIQ", datadir, pheno=pheno)
norm_data_list[['UBIQ']] = ub  

# prot-ph
ph = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ph/normalized-data", 
                             "med-mad-normalized-logratio.txt", "PHOSPHO", datadir, pheno=pheno)
norm_data_list[['PHOSPHO']] = ph
```

```{r save list}
save(norm_data_list, file=sprintf("%s/norm_data_list.RData", datadir), compression_level = 9)
```

```{r save one per dataset}
# save one table per ome
load(file=sprintf("%s/norm_data_list.RData", datadir))

for(ome in names(norm_data_list)){
  
  for(tissue in names(norm_data_list[[ome]])){
    
    norm_data = norm_data_list[[ome]][[tissue]]
    
    # make sure values are numeric
    numeric_cols = colnames(norm_data)[2:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
    
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    new_name = sprintf("%s_%s_NORM_DATA",ome,gsub("-","",tissue))
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))
    
    # Save .rda - this keeps giving me an error early in the loop
    do.call("use_data", list(as.name(new_name), overwrite = TRUE))
  }
}
```

## Handle IMMUNO data separately 
Protein extractions from the same viallabel were used in multiple panels. 
Furthermore, not all assays were performed on every viallabel. 
CHEX* variables are also well-specific measurements, where CHEX4 is used as a covariate in differential analysis. 
```{r immuno}
metadata = dl_read_gcp("gs://mawg-data/pass1b-06/immunoassay/data/release/sample_metadata_20210803.csv", sep=',')
metadata[,group := ifelse(intervention == 'control', 'control', sacrifice_time)]
metadata[,c('intervention','sacrifice_time') := NULL]

# version of the data used for differential analysis 
flist = system("gsutil ls gs://mawg-data/pass1b-06/immunoassay/data/release/pass1b-06*_mfi-log2-filt-imputed-na-outliers.txt", intern=T)

# add CHEX to metadata
new_meta_list = list()

imputed_data = list()

for(f in flist){
  splits = unname(unlist(strsplit(basename(f), '_')))
  panel = splits[4]
  if(!panel %in% metadata[,panel_name] & panel %in% c("ADIPONECTIN","SERPIN-E")){
    meta_panel = "rat-adipokine"
  }else{
    meta_panel = panel
  }
  tissue = splits[2]
  tissue_abbrev = MotrpacBicQC::tissue_abbr[[tissue]]
  print(sprintf("%s %s", panel, tissue))
  if(!panel %in% names(imputed_data)){
    imputed_data[[panel]] = list()
  }
  if(!tissue %in% names(imputed_data[[panel]])){
    imputed_data[[panel]][[tissue_abbrev]] = list()
  }
  # add CHEX to metadata
  imputed = dl_read_gcp(f, sep='\t')
  imputed[,vial_label := as.character(vial_label)]
  new_meta_list[[sprintf("%s;%s", panel, tissue)]] = merge(metadata[tissue_code==tissue & panel_name==meta_panel], 
                                                           imputed[,.(vial_label, CHEX1, CHEX2, CHEX3, CHEX4)],
                                                           by="vial_label")
  # remove CHEX from imputed data
  imputed[,c("CHEX1","CHEX2","CHEX3","CHEX4") := NULL]
  
  # actually don't do this. it makes life harder downstream
  # # format like other omes 
  # imputed_df = as.data.frame(imputed)
  # rownames(imputed_df) = imputed_df$vial_label
  # imputed_df$vial_label = NULL
  # imputed_t = t(imputed_df)
  # imputed_t_dt = data.table(cbind(data.frame(feature=rownames(imputed_t)), imputed_t))
  # 
  # # fix "feature"
  # imputed_t_dt[,feature := sprintf("IMMUNO;%s;%s", tissue_abbrev, feature)]
  # 
  # # save features that were renamed 
  # curr_repeated_feature_map = repeated_feature_map[panel == meta_panel]
  # renamed = imputed_t_dt[feature %in% curr_repeated_feature_map[,feature]]
  # if(nrow(renamed) > 0){
  #   renamed[,feature := curr_repeated_feature_map[match(renamed[,feature], feature), new_feature]]
  #   repeated_feature_data[[sprintf("%s;%s", panel, tissue)]] = renamed
  # }
  
  setnames(imputed, "vial_label", "viallabel")
  imputed = as.data.frame(imputed)
  imputed_data[[panel]][[tissue_abbrev]] = imputed
}
new_meta = rbindlist(new_meta_list)
# add ref stds
metadata[!vial_label %in% new_meta[,vial_label]]
new_meta = rbindlist(list(new_meta, metadata[!vial_label %in% new_meta[,vial_label]]), fill=T)
setnames(new_meta, "vial_label", "viallabel")
new_meta[new_meta==""] = NA

# save data
IMMUNO_META = as.data.frame(new_meta)
usethis::use_data(IMMUNO_META, overwrite=T)
sinew::makeOxygen(IMMUNO_META)

# write metadata to file 
write.table(IMMUNO_META, file=sprintf("%s/sample_metadata_20220713.csv",scratch), sep=",", col.names=T, row.names=F, quote=F)
system(sprintf("gsutil cp %s gs://mawg-data/pass1b-06/immunoassay/data/release/",
               sprintf("%s/sample_metadata_20220713.csv",scratch)))

# save the data in list format 
IMMUNO_NORM_DATA = imputed_data
usethis::use_data(IMMUNO_NORM_DATA, overwrite=T)
```

### Create single feature x pid table for IMMUNO
```{r format IMMUNO for viz}
load("../../data/IMMUNO_NORM_DATA.rda")
load("../../data/REPEATED_FEATURES.rda")
rep = data.table(REPEATED_FEATURES)
names(IMMUNO_NORM_DATA)
data_list = list()
for(panel in names(IMMUNO_NORM_DATA)){
  for(tissue in names(IMMUNO_NORM_DATA[[panel]])){
    df = IMMUNO_NORM_DATA[[panel]][[tissue]]
    # transpose
    rownames(df) = df$viallabel
    df$viallabel = NULL
    df_t = as.data.frame(t(df))
    feature_ID = rownames(df_t)
    # fix repeated features
    features = sprintf("IMMUNO;%s;%s", tissue, feature_ID)
    if(any(features %in% rep[,feature])){
      new_features = sprintf("IMMUNO;%s;%s:%s", tissue, panel, feature_ID)
      features[new_features %in% rep[,new_feature]] = new_features[new_features %in% rep[,new_feature]]
    }
    # replace colname with pid 
    colnames(df_t) = viallabel_to_pid(colnames(df_t))
    # make data.table
    dt = as.data.table(cbind(feature=features, df_t))
    label=sprintf("%s;%s", tissue, panel)
    writeLines(label)
    data_list[[label]] = dt
  }
}
immuno_data = rbindlist(data_list, fill=T)
# make sure rownames are unique
length(immuno_data[,feature]) == length(unique(immuno_data[,feature]))
# check that repeated features match
all(immuno_data[grepl(":",feature),feature] %in% rep[,new_feature])

# save 
IMMUNO_VIZ_DATA = as.data.frame(immuno_data)
usethis::use_data(IMMUNO_VIZ_DATA, overwrite=T)
```

## Save METHYL to GDrive
```{r}
# normalized data for viz
rrbs = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data",
  "normalized-log-M-window.txt", "METHYL", datadir, pheno=pheno)
# rrbs = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", "normalized-log-M-window.txt", "METHYL", datadir, pheno=pheno, features_to_keep = rownames(zs_smoothed)[grepl("METHYL", rownames(zs_smoothed))])
# length(rownames(zs_smoothed)[grepl("METHYL", rownames(zs_smoothed))])
# dim(rrbs)
lapply(rrbs, head)
# remove is_outlier row
rrbs = lapply(rrbs, function(x){
  return(x[2:nrow(x)])
})

METHYL_NORM_DATA = lapply(rrbs, function(norm_data){
  # make sure values are numeric
  numeric_cols = colnames(norm_data)[2:ncol(norm_data)]
  norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
  
  # convert to data.frame for others
  norm_data_df = as.data.frame(norm_data)
  
  return(norm_data_df)
})

save(METHYL_NORM_DATA, file="/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data/METHYL/METHYL_NORM_DATA.rda", compress = "bzip2", compression_level = 9)

# save space
rrbs=NULL
METHYL_NORM_DATA=NULL

# save in Dropbox instead: https://www.dropbox.com/sh/vx7vb77g9vctbjq/AADvbE3QJ2DDsC_5-AGg-P_ia?dl=0

# # format to save
# for(tissue in names(rrbs)){
#     
#   norm_data = rrbs[[tissue]]
#   
#   # make sure values are numeric
#   numeric_cols = colnames(norm_data)[2:ncol(norm_data)]
#   norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
#   
#   # convert to data.frame for others
#   norm_data_df = as.data.frame(norm_data)
#   
#   # if it's more than 500k rows, cut in half
#   if(nrow(norm_data_df)>500000){
#     cutoff = ceiling(nrow(norm_data_df)/2)
#     part1 = norm_data_df[1:cutoff,]
#     part2 = norm_data_df[(cutoff+1):nrow(norm_data_df),]
#     
#     new_name1 = sprintf("%s_%s_NORM_DATA_P1","METHYL",gsub("-","",tissue))
#     writeLines(new_name1)
#     assign(new_name1, part1)
#     print(dim(get(new_name1)))
#     print(head(get(new_name1)))
#     # Save .rda 
#     do.call("use_data", list(as.name(new_name1), overwrite = TRUE))
#     
#     new_name2 = sprintf("%s_%s_NORM_DATA_P2","METHYL",gsub("-","",tissue))
#     writeLines(new_name2)
#     assign(new_name2, part2)
#     print(dim(get(new_name2)))
#     print(head(get(new_name2)))
#     # Save .rda 
#     do.call("use_data", list(as.name(new_name2), overwrite = TRUE))
#     
#   }else{
#     # New name
#     new_name = sprintf("%s_%s_NORM_DATA","METHYL",gsub("-","",tissue))
#     writeLines(new_name)
#     assign(new_name, norm_data_df)
#     
#     print(dim(get(new_name)))
#     print(head(get(new_name)))
#     
#     # Save .rda 
#     do.call("use_data", list(as.name(new_name), overwrite = TRUE))
#   }
# } 
```

```{r methyl raw counts}
rrbs = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data",
  "epigen-rrbs_counts-window.txt", "METHYL", datadir, pheno=pheno)

METHYL_RAW_COUNTS = lapply(rrbs, function(norm_data){
  # remove "is_outlier" row
  norm_data = norm_data[2:nrow(norm_data)]
  
  # make sure values are integer
  numeric_cols = colnames(norm_data)[2:ncol(norm_data)]
  norm_data[,(numeric_cols) := lapply(.SD, as.integer), .SDcols = numeric_cols]
  
  # convert to data.frame for others
  norm_data_df = as.data.frame(norm_data)
  
  return(norm_data_df)
})

save(METHYL_RAW_COUNTS, file="/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data/METHYL/METHYL_RAW_COUNTS.rda", compress = "bzip2", compression_level = 9)

# save space
rrbs=NULL
METHYL_RAW_COUNTS=NULL
```

## Save ATAC to GDrive
```{r atac norm counts}
# atac
atac = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-atac-seq/normalized-data", 
  "quant-norm", "ATAC", datadir, pheno=pheno, tissue_index = 4)
# atac = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-atac-seq/normalized-data", 
#                                "quant-norm", "ATAC", datadir, pheno=pheno, 
#                                features_to_keep = rownames(zs_smoothed)[grepl("ATAC", rownames(zs_smoothed))], tissue_index = 4,
#                                colname_type="viallabel",
#                                return_list = T)
# length(rownames(zs_smoothed)[grepl("ATAC", rownames(zs_smoothed))])
# sum(unlist(lapply(atac, nrow)))
lapply(atac, head)

ATAC_NORM_DATA = lapply(atac, function(norm_data){
  
  # make sure values are numeric
  numeric_cols = colnames(norm_data)[2:ncol(norm_data)]
  norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
  
  # convert to data.frame for others
  norm_data_df = as.data.frame(norm_data)
  
  return(norm_data_df)
})
lapply(ATAC_NORM_DATA, head)

# save in Dropbox instead: https://www.dropbox.com/sh/vx7vb77g9vctbjq/AADvbE3QJ2DDsC_5-AGg-P_ia?dl=0

save(ATAC_NORM_DATA, file="/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data/ATAC/ATAC_NORM_DATA.rda", compress = "bzip2", compression_level = 9)

# save space
atac=NULL
ATAC_NORM_DATA=NULL
```

```{r try downloading from DropBox}
tmpdir = "/oak/stanford/groups/smontgom/nicolerg/tmp"

# Add this function to MotrpacRatTraining6mo

#' Download an RData file from Dropbox
#' 
#' @param target character, public URL specifying an RData file ('.rda' suffix) on Dropbox
#' @param tmpdir character, local path in which to download \code{target}
#' @param redownload boolean, whether or not to download the file if it already exists in \code{tmpdir}
#'
#' @return object contained in \code{target}
#'
#' @examples
#' target = "https://www.dropbox.com/s/rsenv5dgsx0to4i/test.rda?raw=1"
#' obj = load_dropbox_rdata(target, tmpdir)
#' obj = load_dropbox_rdata(target, tmpdir, redownload = T)
load_dropbox_rdata = function(target, tmpdir, redownload = FALSE){
  
  # This is intended to be an internal function only
  accepted_files = c("https://www.dropbox.com/s/rsenv5dgsx0to4i/test.rda?raw=1",
                    "https://www.dropbox.com/s/s6c4otggrkyhh05/ATAC_NORM_DATA.rda?raw=1",
                    "https://www.dropbox.com/s/t92nm792snxzi3c/METHYL_NORM_DATA.rda?raw=1",
                    "https://www.dropbox.com/s/b2wxfvb5q59xtw0/METHYL_RAW_COUNTS.rda?raw=1")
  if(!target %in% accepted_files){
    warning(sprintf("Unrecognized Dropbox link. 'target' is expected to be one of:\n%s",
                    paste0(accepted_files, collapse="\n")))
  }
  
  # Check that it ends in "?raw=1"
  if(!endsWith(target, "?raw=1")){
    stop("'target' should end with '?raw=1'.")
  }
  
  # Check that it ends in ".rda?raw=1"
  if(!endsWith(gsub("\\?raw=1", "", basename(target)), ".rda")){
    stop("'target' should be an RData file with suffix '.rda'.")
  }
  
  dest = sprintf("%s/%s", tmpdir, gsub("\\?raw=1","",basename(target)))
  
  if(!file.exists(dest) | redownload){
    download.file(target,
                  destfile = dest,
                  method = "auto")
  }
  
  if(target=="https://www.dropbox.com/s/s6c4otggrkyhh05/ATAC_NORM_DATA.rda?raw=1"){
    message("Loading and returning an object of size 3.31GB...")
  }else if (target=="https://www.dropbox.com/s/b2wxfvb5q59xtw0/METHYL_RAW_COUNTS.rda?raw=1"){
    message("Loading and returning an object of size 3.70GB...")
  }else if(target=="https://www.dropbox.com/s/t92nm792snxzi3c/METHYL_NORM_DATA.rda?raw=1"){
    message("Loading and returning an object of size 3.70GB...")
  }
  
  load(dest)
  obj_name = gsub("\\.rda","",basename(dest))
  if(!exists(obj_name)){
    stop(sprintf("File '%s' doesn't contain an object called '%s'. 'target' must specify an RData file that contains a single object with the same name as the file.", basename(dest), obj_name))
  }
  
  return(get(obj_name))
}

target = "https://www.dropbox.com/s/rsenv5dgsx0to4i/test.rda?raw=1"
obj = load_dropbox_rdata(target, tmpdir)
obj = load_dropbox_rdata(target, tmpdir, redownload = T)

target = "https://www.dropbox.com/s/s6c4otggrkyhh05/ATAC_NORM_DATA.rda?raw=1"
obj = load_dropbox_rdata(target, tmpdir)
```

## Format METAB sample-level data
### Create single feature x pid table for METAB
```{r}
# start with Pierre's .rda
load("../../data/METAB_SAMPLE_DATA.rda")
data_list = list()
for(i in 1:nrow(METAB_SAMPLE_DATA)){
  tissue_code = METAB_SAMPLE_DATA$tissue[i]
  tissue = TISSUE_CODE_TO_ABBREV[[tissue_code]]
  assay_code = METAB_SAMPLE_DATA$assay_code[i]
  sample_data = as.data.frame(METAB_SAMPLE_DATA$sample_data[i][[1]])
  
  # replace colname with pid 
  colnames(sample_data) = viallabel_to_pid(colnames(sample_data))
  
  # make original and "fixed" feature for ALL feature_ID
  features = sprintf("METAB;%s;%s", tissue, rownames(sample_data))
  new_features = sprintf("METAB;%s;%s:%s", tissue, assay_code, rownames(sample_data))

  # make data.table
  dt = as.data.table(cbind(feature=features, 
                           new_feature=new_features,
                           sample_data))
  label=sprintf("%s;%s", tissue, assay_code)
  writeLines(label)
  data_list[[i]] = dt
}
metab_data = rbindlist(data_list, fill=T)
```

```{r}
# GOAL: get the sample-level data to match `feature` in the states table, as well as `feature_ID` from the sample-level data 
differential_features = GRAPH_STATES$feature

# now, deal with duplicate metabolites
# what are the features present in the DEA tables? the sample-level tables? how to adjust them?
# get meta-reg dea
metab_metareg = dl_load_gcp("gs://motrpac-data-freeze-pass/pass1b-06/v1.0/analysis/metabolomics-named-merged/dea/meta-regression/metab_metareg_20211018.RData", "metab_metareg")
metab_dea = as.data.table(metab_metareg$training_dea)
metab_dea[,feature := sprintf("METAB;%s;%s", tissue_abbreviation, feature_ID)]
diff_metab = differential_features[grepl("METAB;", differential_features)]
table(diff_metab %in% metab_dea[,feature]) # 205 differential metabolite features not in training-dea results 
table(metab_dea[,feature_ID] %in% rep[,feature_ID])
table(diff_metab %in% rep[,new_feature])
# all features in diff_metab but not in training-dea are explained by renaming duplicate metabolites
# features are prepended with dataset, e.g. GMP --> meta-reg:GM, metab-u-ionpneg:GMP, metab-u-rppos:GMP
# in our case, we want ALL individual sample-level data
# do all differential metabolites match if we remove dataset from feature?
head(diff_metab[!diff_metab %in% metab_dea[,feature]])
fixed = gsub(";metab-[a-z-]+:|;meta-reg:",";",diff_metab[!diff_metab %in% metab_dea[,feature]])
table(fixed %in% metab_dea[,feature])
# okay, so all 205 features in diff_metab but not in training-dea are addressed by reverting diff_metab feature_ID back to original feature_ID 
# this means we need to (1) rename duplicate features from sample-level data and (2) handle duplicate feature names WITHIN the plotting function, which will exist because of the meta-regression

# rename feature_ID when duplicated
metab_data[,feature := ifelse(new_feature %in% rep[,new_feature], new_feature, feature)]
metab_data[,new_feature := NULL]

# do we cover all diff metab?
table(diff_metab %in% metab_data[,feature])

# find subset of metab named by refmet instead of metabolite_name. these are hits from meta-reg
metab_data[,metabolite_name := gsub(".*;","",feature)]
metab_data[,refmet := mdd[match(metab_data[,metabolite_name], metabolite_name), refmet_name]]
metab_data_sub = metab_data[!is.na(refmet)]
metab_data_sub[,tissue := unname(unlist(sapply(feature, function(x) unname(unlist(strsplit(x, ';')))[2])))]
metab_data_sub[,new_feature := sprintf("METAB;%s;%s", tissue, refmet)]
metab_data_sub = metab_data_sub[new_feature %in% diff_metab]
metab_data_sub = metab_data_sub[!feature == new_feature]
metab_data_sub[,feature := new_feature]
metab_data_sub[,c('metabolite_name', 'refmet', 'tissue', 'new_feature') := NULL]

# save these. duplicate if necessary 
metab_data2 = rbindlist(list(metab_data, metab_data_sub), fill=T)
metab_data2[,c("metabolite_name","refmet") := NULL]

# what's left?
table(diff_metab %in% metab_data2[,feature])
diff_metab[!diff_metab %in% metab_data2[,feature]]

# add these guys 
# METAB;BAT;meta-reg:GMP
# METAB;LIVER;meta-reg:GMP 
# METAB;LUNG;meta-reg:GMP
# METAB;SKM-GN;meta-reg:GMP
# METAB;WAT-SC;meta-reg:GMP

sub2 = metab_data2[feature %in% c("METAB;BAT;GMP", "METAB;LIVER;GMP", "METAB;LUNG;GMP", "METAB;SKM-GN;GMP", "METAB;WAT-SC;GMP")]
sub2[,feature := gsub("GMP","meta-reg:GMP",feature)]

metab_data3 = rbindlist(list(metab_data2, sub2), fill=T)
table(diff_metab %in% metab_data3[,feature])

# now all differential features exist in metab_data3
# there are still duplicate features from non-differential metab, but that's okay (I think?)
METAB_VIZ_DATA = as.data.frame(metab_data3)
usethis::use_data(METAB_VIZ_DATA, overwrite=T)
```

