---
title: "Download sample-level data"
author: "Nicole Gay"
date: "12/9/2021"
output: html_document
---

```{r setup, include=FALSE}
## WARNING: this code is not self-contained. it requires access to the motrpac-mawg GitHub and gs://mawg-data GCP 

library(data.table)
library(MotrpacBicQC)
library(MotrpacRatTraining6moData)
library(dplyr)
library(testit)
library(usethis)
library(sinew)
library(devtools)
knitr::opts_chunk$set(echo = TRUE)

mawgdir = '/oak/stanford/groups/smontgom/nicolerg/MOTRPAC/MAWG_DATA'
gitdir = "/oak/stanford/groups/smontgom/nicolerg/src/MOTRPAC/"
scratch='/oak/stanford/groups/smontgom/nicolerg/tmp'
datadir = '/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data'
gsutil='~/google-cloud-sdk/bin/gsutil'

source(sprintf("%s/motrpac-mawg/pass1b-06/tools/get_fx.R", gitdir))
source(sprintf("%s/motrpac-mawg/pass1b-06/integrative/clustering/cluster_viz_fx.R", gitdir))
load_all("../../../MotrpacRatTraining6mo/R") # viallabel_to_pid()
```

```{r load data}
# feature_to_universe and 5% FDR tables
# system(sprintf("gsutil -m cp -r gs://mawg-data/pass1b-06/merged %s", datadir))
load(sprintf("%s/merged/zs_smoothed-0.05fdr-0logfc-cluster-input_20211115.RData", mawgdir))
load(sprintf("%s/merged/master_feature_to_gene_20211116.RData", mawgdir))

differential_features = rownames(zs_smoothed)
master_feature_to_gene = NULL # don't need this
head(repeated_feature_map)

# read in pheno data
pheno = dl_format_pheno(scratch, gsutil, parallel=T)
```

```{r fx}
combine_normalized_data = function(gsutil_path, 
                                   pattern, 
                                   ome_abbreviation, 
                                   datadir, 
                                   gsutil_bin = '~/google-cloud-sdk/bin/gsutil', 
                                   pheno = NULL, 
                                   features_to_keep = NULL, 
                                   tissue_index=3, 
                                   skip=1, 
                                   n_rows=Inf,
                                   file_list=NULL, 
                                   viallabel=F, 
                                   add_filename_as_column=F,
                                   return_list=T,
                                   colname_type="viallabel",
                                   rename_features=F,
                                   add_more_columns=T){
  norm_list = list()
  
  if(is.null(file_list)){
    # list normalized data
    file_list = system(sprintf("%s ls %s/*%s*", gsutil_bin, gsutil_path, pattern), intern=T)
    if(length(file_list)==0){
      stop(sprintf("No files found for %s/*%s*", gsutil_path, pattern))
    }
    # check if files already exist
    file_names = paste(sprintf("%s/%s", datadir, ome_abbreviation), basename(file_list), sep="/")
    if(!all(file.exists(file_names))){
      # download those files 
      system(sprintf("mkdir -p %s/%s", datadir, ome_abbreviation))
      system(sprintf("%s -m cp %s/*%s* %s/%s", gsutil_bin, gsutil_path, pattern, datadir, ome_abbreviation))
    }
    file_list = list.files(path=sprintf("%s/%s", datadir, ome_abbreviation), pattern=pattern,  full.names = T)
  }

  # read in and format 
  for(f in file_list){
    if(ome_abbreviation=='ATAC' & grepl("_mssm_", f)) next
    print(basename(f))
    if(colname_type == "pid"){
      curr_norm = fread(f, sep='\t', header=T, skip = skip, nrows=n_rows) # use pid as colname 
      if(colnames(curr_norm)[1]!='pid'){
        warning(sprintf("pid is not where expected in the header for %s. Renaming column.",f))
        colnames(curr_norm)[1] = 'pid'
      }else{
        # remove bid
        curr_norm = curr_norm[2:nrow(curr_norm)]
      }
    }else if(colname_type == "viallabel"){
      curr_norm = fread(f, sep='\t', header=T, skip = 0, nrows=n_rows) 
      colname = "viallabel"
      if(colnames(curr_norm)[1]!='viallabel'){
        warning(sprintf("viallabel is not where expected in the header for %s.",f))
        print(head(curr_norm))
        if(colnames(curr_norm)[1] == "viallabel-Me/Un"){
          colname = "viallabel-Me/Un"
        }else if(colnames(curr_norm)[1] == "chrom"){
          if(ome_abbreviation == "ATAC"){
            # make "feature_ID"
            curr_norm = data.table(cbind(
              feature_ID = sprintf("%s:%s-%s",
                                   curr_norm[,chrom],
                                   curr_norm[,start],
                                   curr_norm[,end]),
              curr_norm[,4:ncol(curr_norm), with=F]
            ))
          }else{
            stop()
          }
        }else{
          stop()
        }
      }
      # remove pid, bid
      if(curr_norm[1,get(colname)] == "pid" & curr_norm[2,get(colname)] == "bid"){
        curr_norm = curr_norm[3:nrow(curr_norm)]
      }
    }

    tissue_code = unname(unlist(strsplit(basename(f), '_')))[tissue_index]
    if(!tissue_code %in% names(tissue_abbr)){
      stop(sprintf("Tissue code not recognized for %s: %s", f, tissue_code))
    }
    tissue_abbrv = tissue_abbr[[tissue_code]]
    
    if(rename_features){
      # fix feature id
      feature_names = sprintf("%s;%s;%s", ome_abbreviation, tissue_abbrv, curr_norm[,get(colname_type)])
      curr_norm[,(colname_type) := NULL]
      curr_norm = cbind(data.table(feature=feature_names), curr_norm)
    }else{
      print(colnames(curr_norm)[1])
      print(colname_type)
      setnames(curr_norm, colnames(curr_norm)[1], "feature_ID")
    }
    
    # replace viallabel with pid
    if(viallabel){
      if(is.null(pheno)) stop("pheno needs to be supplied if viallabel=T")
      # replace viallabel with pid
      print(colnames(curr_norm) %in% as.character(pheno[,viallabel]))
      pheno_sub = unique(pheno[,.(viallabel, pid)])
      colnames(curr_norm)[colnames(curr_norm) %in% as.character(pheno[,viallabel])] = 
        as.character(pheno_sub[match(colnames(curr_norm)[colnames(curr_norm) %in% as.character(pheno[,viallabel])], as.character(viallabel)), pid])
    }
    
    if(add_filename_as_column){
      curr_norm[,origin_file := basename(f)]
    }
    
    # add cols: feature, feature_ID, tissue (tissue abbreviation), assay (assay/ome abbreviation)
    curr_norm = data.table(cbind(feature = sprintf("%s;%s;%s", ome_abbreviation, tissue_abbrv, curr_norm[,feature_ID]),
                                 feature_ID = curr_norm[,feature_ID],
                                 tissue = tissue_abbrv,
                                 assay = ome_abbreviation,
                                 curr_norm[,2:ncol(curr_norm), with=F]))
    
    if(!is.null(features_to_keep)){
      # filter down to these features
      print(dim(curr_norm))
      print(length(features_to_keep))
      curr_norm = curr_norm[feature %in% features_to_keep]
      print(dim(curr_norm))
    }
    
    norm_list[[tissue_abbrv]] = curr_norm
    
  }
  
  if(return_list){
    return(norm_list)
  }
  
  norm = rbindlist(norm_list, fill=T)
  
  if(!is.null(pheno)){
    # check if pids are valid 
    if(colname_type == "pid"){
      if(!all(colnames(norm)[2:ncol(norm)] %in% as.character(pheno[,pid]))){
        warning("Some PIDs not found")
      }
    }
    if(colname_type == "viallabel"){
      if(!all(colnames(norm)[2:ncol(norm)] %in% as.character(pheno[,pid]))){
        warning("Some PIDs not found")
      }
    }
  }
  
  return(norm)
}

viallabel_to_pid = function(viallabels){
  pheno = as.data.table(MotrpacRatTraining6moData::PHENO)
  pheno = unique(pheno[,.(viallabel, pid)])
  pheno = pheno[viallabel %in% as.character(viallabels)]
  vl_to_pid = pheno[,pid]
  names(vl_to_pid) = pheno[,viallabel]
  return(vl_to_pid)
}
```

```{r read in for each ome}
norm_data_list = list()

# rna
norm_data_list[["TRNSCRPT"]] = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/transcriptomics/transcript-rna-seq/normalized-data",
                                                       "normalized-log-cpm", "TRNSCRPT", datadir, pheno=pheno)

# prot-pr
prot = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-pr/normalized-data", 
                               "med-mad-normalized-logratio.txt", "PROT", datadir, pheno=pheno)
norm_data_list[['PROT']] = prot

# prot-ac
ac = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ac/normalized-data", 
                             "med-mad-normalized-logratio.txt", "ACETYL", datadir, pheno=pheno)
norm_data_list[['ACETYL']] = ac

# prot-ub
ub = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ub/normalized-data", 
                             "med-mad-normalized-protein-corrected-logratio.txt", "UBIQ", datadir, pheno=pheno)
norm_data_list[['UBIQ']] = ub  

# prot-ph
ph = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/proteomics-untargeted/prot-ph/normalized-data", 
                             "med-mad-normalized-logratio.txt", "PHOSPHO", datadir, pheno=pheno)
norm_data_list[['PHOSPHO']] = ph
```

## Only differential features for ATAC, RRBS
```{r}
# atac
atac = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-atac-seq/normalized-data", 
                               "quant-norm", "ATAC", datadir, pheno=pheno, 
                               features_to_keep = differential_features[grepl("ATAC", differential_features)], tissue_index = 4)
length(differential_features[grepl("ATAC", differential_features)])
sum(unlist(lapply(atac, nrow)))
norm_data_list[['ATAC']] = atac

# rrbs
rrbs = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", 
                               "normalized-log-M-window.txt", "METHYL", datadir, pheno=pheno, 
                               features_to_keep = differential_features[grepl("METHYL", differential_features)])
length(differential_features[grepl("METHYL", differential_features)])
sum(unlist(lapply(rrbs, nrow)))
norm_data_list[['METHYL']] = rrbs
```

```{r save list}
save(norm_data_list, file=sprintf("%s/norm_data_list.RData", datadir))
```

```{r save one per dataset}
# save one table per ome and tissue
load(file=sprintf("%s/norm_data_list.RData", datadir))

for(ome in names(norm_data_list)){
  
  for(tissue in names(norm_data_list[[ome]])){
    
    norm_data = norm_data_list[[ome]][[tissue]]
    
    # make sure values are numeric
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
    
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    if(ome %in% c("ATAC","METHYL")){
      new_name = sprintf("%s_%s_NORM_DATA_05FDR",ome,gsub("-","",tissue))
    }else{
      new_name = sprintf("%s_%s_NORM_DATA",ome,gsub("-","",tissue))
    }
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))
    
    # Save .rda - this keeps giving me an error early in the loop
    do.call("use_data", list(as.name(new_name), overwrite = TRUE))
  }
}
```
## Handle IMMUNO data separately 
Protein extractions from the same viallabel were used in multiple panels. 
Furthermore, not all assays were performed on every viallabel. 
CHEX* variables are also well-specific measurements, where CHEX4 is used as a covariate in differential analysis. 
```{r immuno}
metadata = dl_read_gcp("gs://mawg-data/pass1b-06/immunoassay/data/release/sample_metadata_20210803.csv", sep=',')
metadata[,group := ifelse(intervention == 'control', 'control', sacrifice_time)]
metadata[,c('intervention','sacrifice_time') := NULL]

# version of the data used for differential analysis 
flist = system("gsutil ls gs://mawg-data/pass1b-06/immunoassay/data/release/pass1b-06*_mfi-log2-filt-imputed-na-outliers.txt", intern=T)

# add CHEX to metadata
new_meta_list = list()

imputed_data = list()

for(f in flist){
  splits = unname(unlist(strsplit(basename(f), '_')))
  panel = splits[4]
  if(!panel %in% metadata[,panel_name] & panel %in% c("ADIPONECTIN","SERPIN-E")){
    meta_panel = "rat-adipokine"
  }else{
    meta_panel = panel
  }
  tissue = splits[2]
  tissue_abbrev = MotrpacBicQC::tissue_abbr[[tissue]]
  print(sprintf("%s %s", panel, tissue))
  if(!panel %in% names(imputed_data)){
    imputed_data[[panel]] = list()
  }
  if(!tissue %in% names(imputed_data[[panel]])){
    imputed_data[[panel]][[tissue_abbrev]] = list()
  }
  # add CHEX to metadata
  imputed = dl_read_gcp(f, sep='\t')
  imputed[,vial_label := as.character(vial_label)]
  new_meta_list[[sprintf("%s;%s", panel, tissue)]] = merge(metadata[tissue_code==tissue & panel_name==meta_panel], 
                                                           imputed[,.(vial_label, CHEX1, CHEX2, CHEX3, CHEX4)],
                                                           by="vial_label")
  # remove CHEX from imputed data
  imputed[,c("CHEX1","CHEX2","CHEX3","CHEX4") := NULL]
  
  setnames(imputed, "vial_label", "viallabel")
  imputed = as.data.frame(imputed)
  imputed_data[[panel]][[tissue_abbrev]] = imputed
}
new_meta = rbindlist(new_meta_list)
# add ref stds
metadata[!vial_label %in% new_meta[,vial_label]]
new_meta = rbindlist(list(new_meta, metadata[!vial_label %in% new_meta[,vial_label]]), fill=T)
setnames(new_meta, "vial_label", "viallabel")
new_meta[new_meta==""] = NA

# save data
IMMUNO_META = as.data.frame(new_meta)
usethis::use_data(IMMUNO_META, overwrite=T)
sinew::makeOxygen(IMMUNO_META)

# write metadata to file 
write.table(IMMUNO_META, file=sprintf("%s/sample_metadata_20220713.csv",scratch), sep=",", col.names=T, row.names=F, quote=F)
system(sprintf("gsutil cp %s gs://mawg-data/pass1b-06/immunoassay/data/release/",
               sprintf("%s/sample_metadata_20220713.csv",scratch)))

# save the data in list format 
IMMUNO_NORM_DATA_NESTED = imputed_data
usethis::use_data(IMMUNO_NORM_DATA_NESTED, overwrite=T)
```

### Create feature x pid tables for IMMUNO
```{r format IMMUNO for viz}
load("../../data/IMMUNO_NORM_DATA_NESTED.rda")
load("../../data/REPEATED_FEATURES.rda")
rep = data.table(REPEATED_FEATURES)
names(IMMUNO_NORM_DATA_NESTED)
data_list = list()
for(panel in names(IMMUNO_NORM_DATA_NESTED)){
  for(tissue in names(IMMUNO_NORM_DATA_NESTED[[panel]])){
    df = IMMUNO_NORM_DATA_NESTED[[panel]][[tissue]]
    # remove refstd
    df = df[!grepl("^8", df$viallabel),]
    # transpose
    rownames(df) = df$viallabel
    df$viallabel = NULL
    df_t = as.data.frame(t(df))
    feature_ID = rownames(df_t)
    # fix repeated features
    features = sprintf("IMMUNO;%s;%s", tissue, feature_ID)
    if(any(features %in% rep[,feature])){
      new_features = sprintf("IMMUNO;%s;%s:%s", tissue, panel, feature_ID)
      features[new_features %in% rep[,new_feature]] = new_features[new_features %in% rep[,new_feature]]
    }
    # replace colname with pid 
    colnames(df_t) = viallabel_to_pid(colnames(df_t))
    # make data.table
    dt = as.data.table(cbind(feature=features, 
                             feature_ID=feature_ID,
                             tissue=tissue, 
                             assay="IMMUNO",
                             dataset=panel,
                             df_t))
    label=sprintf("%s;%s", tissue, panel)
    writeLines(label)
    data_list[[label]] = dt
  }
}
immuno_data = rbindlist(data_list, fill=T)
# make sure rownames are unique
length(immuno_data[,feature]) == length(unique(immuno_data[,feature]))
# check that repeated features match
all(immuno_data[grepl(":",feature),feature] %in% rep[,new_feature])
# remove ref std

# save 
IMMUNO_NORM_DATA_FLAT = as.data.frame(immuno_data)
usethis::use_data(IMMUNO_NORM_DATA_FLAT, overwrite=T)
```

<!-- ## Save METHYL to GCP bucket -->
<!-- ```{r} -->
<!-- # normalized data for viz -->
<!-- rrbs = combine_normalized_data( -->
<!--   "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", -->
<!--   "normalized-log-M-window.txt", "METHYL", datadir, pheno=pheno) -->
<!-- # rrbs = combine_normalized_data("gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", "normalized-log-M-window.txt", "METHYL", datadir, pheno=pheno, features_to_keep = rownames(zs_smoothed)[grepl("METHYL", rownames(zs_smoothed))]) -->
<!-- # length(rownames(zs_smoothed)[grepl("METHYL", rownames(zs_smoothed))]) -->
<!-- # dim(rrbs) -->
<!-- lapply(rrbs, head) -->
<!-- # remove is_outlier row -->
<!-- rrbs = lapply(rrbs, function(x){ -->
<!--   return(x[2:nrow(x)]) -->
<!-- }) -->

<!-- METHYL_NORM_DATA = lapply(rrbs, function(norm_data){ -->
<!--   # make sure values are numeric -->
<!--   numeric_cols = colnames(norm_data)[2:ncol(norm_data)] -->
<!--   norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols] -->

<!--   # convert to data.frame for others -->
<!--   norm_data_df = as.data.frame(norm_data) -->

<!--   return(norm_data_df) -->
<!-- }) -->

<!-- save(METHYL_NORM_DATA, file="/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data/METHYL/METHYL_NORM_DATA.rda", compress = "bzip2", compression_level = 9) -->

<!-- # save space -->
<!-- rrbs=NULL -->
<!-- METHYL_NORM_DATA=NULL -->

<!-- # save in Dropbox instead: https://www.dropbox.com/sh/vx7vb77g9vctbjq/AADvbE3QJ2DDsC_5-AGg-P_ia?dl=0 -->
<!-- ``` -->

<!-- ```{r} -->
<!-- # write one file per tissue instead  -->
<!-- load("/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data/METHYL/METHYL_NORM_DATA.rda") -->
<!-- #TODO -->
<!-- ``` -->

<!-- ```{r methyl raw counts} -->
<!-- rrbs = combine_normalized_data( -->
<!--   "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-rrbs/normalized-data", -->
<!--   "epigen-rrbs_counts-window.txt", "METHYL", datadir, pheno=pheno) -->

<!-- METHYL_RAW_COUNTS = lapply(rrbs, function(norm_data){ -->
<!--   # remove "is_outlier" row -->
<!--   norm_data = norm_data[2:nrow(norm_data)] -->

<!--   # make sure values are integer -->
<!--   numeric_cols = colnames(norm_data)[2:ncol(norm_data)] -->
<!--   norm_data[,(numeric_cols) := lapply(.SD, as.integer), .SDcols = numeric_cols] -->

<!--   # convert to data.frame for others -->
<!--   norm_data_df = as.data.frame(norm_data) -->

<!--   return(norm_data_df) -->
<!-- }) -->

<!-- save(METHYL_RAW_COUNTS, file="/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data/METHYL/METHYL_RAW_COUNTS.rda", compress = "bzip2", compression_level = 9) -->

<!-- # save space -->
<!-- rrbs=NULL -->
<!-- METHYL_RAW_COUNTS=NULL -->
<!-- ``` -->

<!-- ```{r} -->
<!-- # write one file per tissue instead  -->
<!-- load("/oak/stanford/groups/smontgom/shared/motrpac/internal_releases/motrpac-data-freeze-pass/v1.1/extracted_sample_level_data/METHYL/METHYL_RAW_COUNTS.rda") -->
<!-- #TODO -->
<!-- ``` -->

## Save ATAC to GDrive
```{r atac norm counts}
# atac
atac = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/analysis/epigenomics/epigen-atac-seq/normalized-data", 
  "quant-norm", "ATAC", datadir, pheno=pheno, tissue_index = 4)

# write one file per tissue
ome="ATAC"
for(tissue in names(atac)){
  
    norm_data = atac[[tissue]]
    
    # make sure values are numeric
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.numeric), .SDcols = numeric_cols]
    
    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    new_name = sprintf("%s_%s_NORM_DATA",ome,gsub("-","",tissue))
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    outfile = sprintf("%s/ATAC/%s.rda", datadir, new_name)
    print(outfile)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))

    # save rda
    do.call("save", list(as.name(new_name), 
                         file = outfile, 
                         compress = "bzip2", compression_level = 9))
}
# save space
atac=NULL
```

```{r ATAC raw counts}
# atac
atac = combine_normalized_data(
  "gs://motrpac-data-freeze-pass/pass1b-06/v1.1/results/epigenomics/*/epigen-atac-seq", 
  "epigen-atac-seq_counts", "ATAC", datadir, pheno=pheno, tissue_index = 3)

# write one file per tissue
ome="ATAC"
for(tissue in names(atac)){
  
    norm_data = atac[[tissue]]
    
    # remove refstd
    to_remove = colnames(norm_data)[grepl("^8", colnames(norm_data))]
    if(length(to_remove)>0){
      norm_data[,c(to_remove) := NULL]
    }
    
    # make sure values are integer
    numeric_cols = colnames(norm_data)[5:ncol(norm_data)]
    norm_data[,(numeric_cols) := lapply(.SD, as.integer), .SDcols = numeric_cols]

    # convert to data.frame for others
    norm_data_df = as.data.frame(norm_data)
    
    # New name
    new_name = sprintf("%s_%s_RAW_COUNTS",ome,gsub("-","",tissue))
    writeLines(new_name)
    assign(new_name, norm_data_df)
    
    outfile = sprintf("%s/ATAC/%s.rda", datadir, new_name)
    print(outfile)
    
    print(dim(get(new_name)))
    print(head(get(new_name)))

    # save rda
    do.call("save", list(as.name(new_name), 
                         file = outfile, 
                         compress = "bzip2", compression_level = 9))
}
# save space
atac=NULL
```

## Format METAB sample-level data
### Create single feature x pid table for METAB
```{r collect metab data}
# start with Pierre's .rda
load("../../data/METAB_SAMPLE_DATA.rda")
data_list = list()
feature_meta_list = list()
for(i in 1:nrow(METAB_SAMPLE_DATA)){
  tissue_code = METAB_SAMPLE_DATA$tissue[i]
  tissue = TISSUE_CODE_TO_ABBREV[[tissue_code]]
  assay_code = METAB_SAMPLE_DATA$assay_code[i]
  sample_data = as.data.frame(METAB_SAMPLE_DATA$sample_data[i][[1]])
  
  # replace colname with pid 
  colnames(sample_data) = viallabel_to_pid(colnames(sample_data))
  feature_IDs = rownames(sample_data)
  
  # make original and "fixed" feature for ALL feature_ID
  features = sprintf("METAB;%s;%s", tissue, feature_IDs)
  new_features = sprintf("METAB;%s;%s:%s", tissue, assay_code, feature_IDs)

  # make data.table
  dt = as.data.table(cbind(feature=features, 
                           new_feature=new_features, 
                           feature_ID=feature_IDs,
                           tissue=tissue,
                           assay="METAB",
                           dataset=assay_code,

                           sample_data))
  label=sprintf("%s;%s", tissue, assay_code)

  data_list[[i]] = dt
  
  feature_meta_list[[i]] = data.table(cbind(tissue = METAB_SAMPLE_DATA[i, "tissue"],
                                    assay_code = METAB_SAMPLE_DATA[i, "assay_code"],
                                    as.data.table(METAB_SAMPLE_DATA[i, "feature_metadata"][[1]])))
  writeLines(label)
  
}
metab_data = rbindlist(data_list, fill=T)
feature_meta = rbindlist(feature_meta_list)

table(unique(metab_data[,feature_ID]) %in% feature_meta[,metabolite_name])
metab_data[!feature_ID %in% feature_meta[,metabolite_name]]
feature_meta[grepl("TG(36:1)>TG(4:0_16:0_16:1)_and_TG(2:0_16:0_18:1)_feature4",metabolite_name)]
# one metabolite doesn't have feature metadata: TG(36:1)>TG(4:0_16:0_16:1)_and_TG(2:0_16:0_18:1)_feature4
# present in DEA though: TG(36:1)>TG(4:0_16:0_16:1)_and_TG(2:0_16:0_18:1)_feature4

# what's the overlap between feature_meta and MotrpacBicQC::metabolomics_data_dictionary?
mdd = data.table(MotrpacBicQC::metabolomics_data_dictionary)
table(unique(feature_meta[,refmet_name]) %in% mdd[,refmet_name])
# only 97 features don't overlap by refmet_name
# do these exist in CURRENT_REFMET_NAME?
not_found = unique(na.omit(feature_meta[!refmet_name %in% mdd[,refmet_name],refmet_name]))
# not_found
#  [1] "CAR(4:0)_rp_a"  "CAR(5:0)_rp_a"  "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "LPC(16:0)_rp_b" "LPC(18:1)_rp_b" "LPC(18:2)_rp_b" "LPC(20:3)_rp_b"
#  [9] "LPC(20:4)_rp_b" "LPE(18:0)_rp_b" "LPE(18:2)_rp_b" "MG(18:1)_rp_b"  "CAR(4:0)_rp_a"  "CAR(5:0)_rp_a"  "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"
# [17] "LPC(15:0)_rp_a" "LPC(16:0)_rp_b" "LPC(18:1)_rp_b" "LPC(18:2)_rp_b" "LPC(20:0)_rp_a" "LPC(20:3)_rp_b" "LPC(20:4)_rp_b" "LPE(18:0)_rp_b"
# [25] "LPE(18:2)_rp_b" "MG(14:0)_rp_a"  "MG(18:1)_rp_b"  "CAR(4:0)_rp_a"  "CAR(5:0)_rp_a"  "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "CAR(4:0)_rp_a"
# [33] "CAR(5:0)_rp_a"  "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "CAR(4:0)_rp_a"  "CAR(5:0)_rp_a"  "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "LPC(16:0)_rp_b"
# [41] "LPC(18:1)_rp_b" "LPC(18:2)_rp_b" "LPC(20:3)_rp_b" "LPC(20:4)_rp_b" "LPE(18:0)_rp_b" "LPE(18:2)_rp_b" "MG(18:1)_rp_b"  "CAR(4:0)_rp_a"
# [49] "CAR(5:0)_rp_a"  "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "LPC(15:0)_rp_a" "LPC(16:0)_rp_b" "LPC(18:1)_rp_b" "LPC(18:2)_rp_b" "LPC(20:0)_rp_a"
# [57] "LPC(20:3)_rp_b" "LPC(20:4)_rp_b" "LPE(18:0)_rp_b" "LPE(18:2)_rp_b" "MG(14:0)_rp_a"  "MG(18:1)_rp_b"  "CAR(4:0)_rp_a"  "CAR(5:0)_rp_a"
# [65] "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "LPC(16:0)_rp_b" "LPC(18:1)_rp_b" "LPC(18:2)_rp_b" "LPC(20:3)_rp_b" "LPC(20:4)_rp_b" "LPE(18:0)_rp_b"
# [73] "LPE(18:2)_rp_b" "MG(18:1)_rp_b"  "CAR(4:0)_rp_a"  "CAR(5:0)_rp_a"  "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "LPC(16:0)_rp_b" "LPC(18:1)_rp_b"
# [81] "LPC(18:2)_rp_b" "LPC(20:3)_rp_b" "LPC(20:4)_rp_b" "LPE(18:0)_rp_b" "LPE(18:2)_rp_b" "MG(18:1)_rp_b"  "CAR(4:0)_rp_a"  "CAR(5:0)_rp_a"
# [89] "CAR(4:0)_rp_b"  "CAR(5:0)_rp_b"  "LPC(18:1)_rp_b" "LPC(18:2)_rp_b" "LPC(20:3)_rp_b" "LPC(20:4)_rp_b" "LPE(18:0)_rp_b" "MG(18:1)_rp_b"
table(not_found %in% mdd[,CURRENT_REFMET_NAME])

# what is the overlap between feature_meta and Metabolomics Workbench?
mwb = data.table(get_and_validate_mdd())
table(unique(feature_meta[,refmet_name]) %in% mwb[,refmet_name])
# not good

# just use feature_meta as metabolomics data dictionary for now 
```

```{r make metab match}
# we have 3 sets of metab data
# GOAL: be able to map metabolites across data sets
# Dataset 1: sample-level data, "metab_data"
# - original feature_ID from sites ("metabolite_name" in `feature_meta`)
# Dataset 2: differential analysis results 
# - mixture of original feature_ID from sites AND refmet IDs ("metabolite_name" and "refmet_name" in `feature_meta`)
# - mixture is from renaming redundant feature_ID by refmet_name
# Dataset 3: graphical analysis of differential features
# - mixture of names from (2) AND renamed feature_IDs documents in REPEATED_FEATURES 

## get datasets 
# Dataset 1
head(metab_data)
# Dataset 2
metab_metareg = dl_load_gcp("gs://motrpac-data-freeze-pass/pass1b-06/v1.0/analysis/metabolomics-named-merged/dea/meta-regression/metab_metareg_20211018.RData", "metab_metareg")
metab_dea = as.data.table(metab_metareg$training_dea)
metab_dea[,feature := sprintf("METAB;%s;%s", tissue_abbreviation, feature_ID)]
metab_dea[,tissue_code := tissue]
metab_dea[,tissue := TISSUE_CODE_TO_ABBREV[tissue_code]]
metab_dea = unique(metab_dea[,.(feature_ID, tissue, dataset, metabolite_refmet, metabolite)])
# Dataset 3
differential_features = MotrpacRatTraining6moData::GRAPH_STATES$feature
diff_metab = differential_features[grepl("METAB;", differential_features)]
# get repeated training-regulated features
# this documents feature_IDs that were renamed for graphical analysis due to duplicate measurements 
rep = data.table(MotrpacRatTraining6moData::REPEATED_FEATURES)

# keep track of all feature IDs in sample_data 
metab_feature_names = unique(metab_data[,.(feature_ID, dataset, tissue)])
setnames(metab_feature_names, "feature_ID", "sample_data_ID")
nrow(metab_feature_names)
metab_feature_names = merge(metab_feature_names, mdd, by.x="sample_data_ID", by.y="metabolite_name", all.x=TRUE)
nrow(metab_feature_names)
# some metabolites have multiple refmet names?

# first, match meta-regression results 
metab_dea[,unique_feature := sprintf("%s;%s;%s", dataset, tissue, feature_ID)]
metab_feature_names[,unique_feature := sprintf("%s;%s;%s", dataset, tissue, sample_data_ID)]
metab_feature_names[,metaregression_ID := NA_character_]
# features that match to sample data feature_ID:
metab_feature_names[unique_feature %in% metab_dea[,unique_feature], metaregression_ID := sample_data_ID]
# features that match to refmet:
metab_dea_sub = metab_dea[!unique_feature %in% metab_feature_names[,unique_feature]]
table(metab_dea_sub[,feature_ID] %in% metab_feature_names[,refmet_name]) # 1013/1014
# except: TG(36:1)>TG(4:0_16:0_16:1)_and_TG(2:0_16:0_18:1)_feature4 WAT-SC
table(metab_dea_sub[,feature_ID] == metab_dea_sub[,metabolite_refmet]) # 1013/1013 (1 NA, feature above)

metab_feature_names1 = metab_feature_names[!is.na(metaregression_ID)]
metab_feature_names2 = metab_feature_names[is.na(metaregression_ID)]
metab_feature_names2_merged = merge(metab_feature_names2, metab_dea_sub, 
                                    by.x=c("tissue","dataset","refmet_name"),
                                    by.y=c("tissue","dataset","feature_ID"))
# ??????
table(unique(metab_dea[,metabolite_refmet]) %in% metab_feature_names2[,refmet_name])
table(unique(metab_dea[,metabolite_refmet]) %in% mdd[,refmet_name])
# LOL 


# original code
# # GOAL: get the sample-level data to match `feature` in the states table, as well as `feature_ID` from the sample-level data 
# differential_features = GRAPH_STATES$feature
# # now, deal with duplicate metabolites
# # what are the features present in the DEA tables? the sample-level tables? how to adjust them?
# # get meta-reg dea
# metab_metareg = dl_load_gcp("gs://motrpac-data-freeze-pass/pass1b-06/v1.0/analysis/metabolomics-named-merged/dea/meta-regression/metab_metareg_20211018.RData", "metab_metareg")
# metab_dea = as.data.table(metab_metareg$training_dea)
# metab_dea[,feature := sprintf("METAB;%s;%s", tissue_abbreviation, feature_ID)]
# diff_metab = differential_features[grepl("METAB;", differential_features)]
# table(diff_metab %in% metab_dea[,feature]) # 205 differential metabolite features not in training-dea results 
# table(metab_dea[,feature_ID] %in% rep[,feature_ID])
# table(diff_metab %in% rep[,new_feature])
# # all features in diff_metab but not in training-dea are explained by renaming duplicate metabolites
# # features are prepended with dataset, e.g. GMP --> meta-reg:GM, metab-u-ionpneg:GMP, metab-u-rppos:GMP
# # in our case, we want ALL individual sample-level data
# # do all differential metabolites match if we remove dataset from feature?
# head(diff_metab[!diff_metab %in% metab_dea[,feature]])
# fixed = gsub(";metab-[a-z-]+:|;meta-reg:",";",diff_metab[!diff_metab %in% metab_dea[,feature]])
# table(fixed %in% metab_dea[,feature])
# # okay, so all 205 features in diff_metab but not in training-dea are addressed by reverting diff_metab feature_ID back to original feature_ID 
# # this means we need to (1) rename duplicate features from sample-level data and (2) handle duplicate feature names WITHIN the plotting function, which will exist because of the meta-regression
# # rename feature_ID when duplicated
# metab_data[,feature := ifelse(new_feature %in% rep[,new_feature], new_feature, feature)]
# metab_data[,new_feature := NULL]
# # do we cover all diff metab?
# table(diff_metab %in% metab_data[,feature])
# # find subset of metab named by refmet instead of metabolite_name. these are hits from meta-reg
# metab_data[,metabolite_name := gsub(".*;","",feature)]
# metab_data[,refmet := mdd[match(metab_data[,metabolite_name], metabolite_name), refmet_name]]
# metab_data_sub = metab_data[!is.na(refmet)]
# metab_data_sub[,tissue := unname(unlist(sapply(feature, function(x) unname(unlist(strsplit(x, ';')))[2])))]
# metab_data_sub[,new_feature := sprintf("METAB;%s;%s", tissue, refmet)]
# metab_data_sub = metab_data_sub[new_feature %in% diff_metab]
# metab_data_sub = metab_data_sub[!feature == new_feature]
# metab_data_sub[,feature := new_feature]
# metab_data_sub[,c('metabolite_name', 'refmet', 'tissue', 'new_feature') := NULL]
# # save these. duplicate if necessary 
# metab_data2 = rbindlist(list(metab_data, metab_data_sub), fill=T)
# metab_data2[,c("metabolite_name","refmet") := NULL]
# # what's left?
# table(diff_metab %in% metab_data2[,feature])
# diff_metab[!diff_metab %in% metab_data2[,feature]]
# # add these guys 
# # METAB;BAT;meta-reg:GMP
# # METAB;LIVER;meta-reg:GMP 
# # METAB;LUNG;meta-reg:GMP
# # METAB;SKM-GN;meta-reg:GMP
# # METAB;WAT-SC;meta-reg:GMP
# sub2 = metab_data2[feature %in% c("METAB;BAT;GMP", "METAB;LIVER;GMP", "METAB;LUNG;GMP", "METAB;SKM-GN;GMP", "METAB;WAT-SC;GMP")]
# sub2[,feature := gsub("GMP","meta-reg:GMP",feature)]
# metab_data3 = rbindlist(list(metab_data2, sub2), fill=T)
# table(diff_metab %in% metab_data3[,feature])
# # now all differential features exist in metab_data3
# # there are still duplicate features from non-differential metab, but that's okay (I think?)
# METAB_VIZ_DATA = as.data.frame(metab_data3)
# usethis::use_data(METAB_VIZ_DATA, overwrite=T)
```

