<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="The goal of the Molecular Transducers of Physical Activity
    Consortium (MoTrPAC) is to study the molecular changes that occur
    during and after exercise and ultimately advance the understanding of
    how physical activity improves and preserves health. This package
    provides convenient access to the multi-tissue, multi-omic data
    generated by MoTrPAC's endurance exercise training study in
    6-month-old rats.">
<title>Data for analysis of the MoTrPAC endurance exercise training study
    in 6-month-old rats • MotrpacRatTraining6moData</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Data for analysis of the MoTrPAC endurance exercise training study
    in 6-month-old rats">
<meta property="og:description" content="The goal of the Molecular Transducers of Physical Activity
    Consortium (MoTrPAC) is to study the molecular changes that occur
    during and after exercise and ultimately advance the understanding of
    how physical activity improves and preserves health. This package
    provides convenient access to the multi-tissue, multi-omic data
    generated by MoTrPAC's endurance exercise training study in
    6-month-old rats.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/tutorial.html">Tutorial for MotrpacRatTraining6moData R package</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="motrpacrattraining6modata">MotrpacRatTraining6moData<a class="anchor" aria-label="anchor" href="#motrpacrattraining6modata"></a>
</h1></div>
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ul>
<li>
<a href="#overview">Overview</a>
<ul>
<li><a href="#about-this-package">About this package</a></li>
<li><a href="#about-motrpac">About MoTrPAC</a></li>
</ul>
</li>
<li>
<a href="#installation">Installation</a>
<ul>
<li><a href="#troubleshooting">Troubleshooting</a></li>
<li><a href="#last-resort">Last resort</a></li>
</ul>
</li>
<li><a href="#explore-available-data-objects">Explore available data objects</a></li>
<li><a href="#access-epigenomics-data-through-google-cloud-storage">Access epigenomics data through Google Cloud Storage</a></li>
<li><a href="#getting-help">Getting help</a></li>
<li><a href="#acknowledgements">Acknowledgements</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<div class="section level3">
<h3 id="about-this-package">About this package<a class="anchor" aria-label="anchor" href="#about-this-package"></a>
</h3>
<p>This package provides convenient access to the processed data and downstream analysis results presented in the main paper for the first large-scale multi-omic multi-tissue endurance exercise training study conducted in young adult rats by the Molecular Transducers of Physical Activity Consortium (MoTrPAC). <em>A <a href="https://www.biorxiv.org/" class="external-link">bioRxiv</a> link to the corresponding preprint will be added shortly.</em></p>
<p>While the data in this package can be used by themselves, the <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo" class="external-link">MotrpacRatTraining6mo</a> R package relies heavily on this package and provides many functions to help retrieve and explore the data. <em>A link to the MotrpacRatTraining6mo vignette will be added shortly.</em></p>
<p><em>A link to the MotrpacRatTraining6moData vignette will be added shortly.</em></p>
</div>
<div class="section level3">
<h3 id="about-motrpac">About MoTrPAC<a class="anchor" aria-label="anchor" href="#about-motrpac"></a>
</h3>
<p>MoTrPAC is a national research consortium designed to discover and perform preliminary characterization of the range of molecular transducers (the “molecular map”) that underlie the effects of physical activity in humans. The program’s goal is to study the molecular changes that occur during and after exercise and ultimately to advance the understanding of how physical activity improves and preserves health. The six-year program is the largest targeted NIH investment of funds into the mechanisms of how physical activity improves health and prevents disease. See <a href="https://www.motrpac.org/" class="external-link">motrpac.org</a> and <a href="https://motrpac-data.org/" class="external-link">motrpac-data.org</a> for more details.</p>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install this package with <code>devtools</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MoTrPAC/MotrpacRatTraining6moData"</span><span class="op">)</span></span></code></pre></div>
<p>The output for a successful installation looks something like this. Note that the <code>*** moving datasets to lazyload DB</code> step takes the longest (~5 minutes):</p>
<pre><code>Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
✓  checking for file ‘.../MoTrPAC-MotrpacRatTraining6moData-1c6478a/DESCRIPTION’ ...
─  preparing ‘MotrpacRatTraining6moData’:
✓  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘MotrpacRatTraining6moData_1.3.2.tar.gz’ (1.3s)
   
* installing *source* package ‘MotrpacRatTraining6moData’ ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (MotrpacRatTraining6moData)</code></pre>
<div class="section level3">
<h3 id="troubleshooting">Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h3>
<p>If you get this error:</p>
<pre><code>Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD' failed</code></pre>
<p>Try extending the timeout:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="fl">1e5</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MoTrPAC/MotrpacRatTraining6moData"</span><span class="op">)</span></span></code></pre></div>
<p>If you get this error after extending the <code>timeout</code>:</p>
<pre><code>Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD' failed
Error in `action()`:
! `class` is absent but must be supplied.
Run `rlang::last_error()` to see where the error occurred.</code></pre>
<p>…this seems to be an intermittent issue seen only on Mac, not Linux or Windows. This was resolved by <a href="https://cran.rstudio.com/" class="external-link">installing the newest version of R</a>.</p>
</div>
<div class="section level3">
<h3 id="last-resort">Last resort<a class="anchor" aria-label="anchor" href="#last-resort"></a>
</h3>
<p>If you can’t get <code>devtools::install_github("MoTrPAC/MotrpacRatTraining6moData")</code> to work, try this:</p>
<ol style="list-style-type: decimal">
<li><p>Go to <a href="https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD" class="external-link uri">https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD</a>, which will automatically start downloading this repository in a tarball</p></li>
<li>
<p>Install the package from source:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"~/Downloads/MoTrPAC-MotrpacRatTraining6moData-0729e2e.tar.gz"</span>, </span>
<span>  repos <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>  type <span class="op">=</span> <span class="st">"source"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData" class="external-link">MotrpacRatTraining6moData</a></span><span class="op">)</span></span></code></pre></div>
</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="explore-available-data-objects">Explore available data objects<a class="anchor" aria-label="anchor" href="#explore-available-data-objects"></a>
</h2>
<p>List available data objects after loading the package:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData" class="external-link">MotrpacRatTraining6moData</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"MotrpacRatTraining6moData"</span><span class="op">)</span></span></code></pre></div>
<p>Get the documentation for a given data object:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">TRNSCRPT_LIVER_DA</span></span></code></pre></div>
<p>Load a data object into your environment using <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">TRNSCRPT_LIVER_DA</span><span class="op">)</span></span></code></pre></div>
<p>List of current objects:</p>
<pre><code>ACETYL_HEART_DA                                Differential analysis of proteomics datasets
ACETYL_HEART_NORM_DATA                         Normalized protein acetylation data
ACETYL_LIVER_DA                                Differential analysis of proteomics datasets
ACETYL_LIVER_NORM_DATA                         Normalized protein acetylation data
ACETYL_META                                    Proteomics sample-level metadata
ASSAY_ABBREV                                   Assay or "ome" abbreviations
ASSAY_ABBREV_TO_CODE                           Assay abbreviation-to-code mapping
ASSAY_CODE_TO_ABBREV                           Assay code-to-abbreviation mapping
ASSAY_COLORS                                   Assay colors
ASSAY_ORDER                                    Assay order
ATAC_BAT_NORM_DATA_05FDR                       Normalized ATAC-seq data for training-regulated features
ATAC_HEART_NORM_DATA_05FDR                     Normalized ATAC-seq data for training-regulated features
ATAC_HIPPOC_NORM_DATA_05FDR                    Normalized ATAC-seq data for training-regulated features
ATAC_KIDNEY_NORM_DATA_05FDR                    Normalized ATAC-seq data for training-regulated features
ATAC_LIVER_NORM_DATA_05FDR                     Normalized ATAC-seq data for training-regulated features
ATAC_LUNG_NORM_DATA_05FDR                      Normalized ATAC-seq data for training-regulated features
ATAC_META                                      ATAC-seq metadata and QC
ATAC_SKMGN_NORM_DATA_05FDR                     Normalized ATAC-seq data for training-regulated features
ATAC_WATSC_NORM_DATA_05FDR                     Normalized ATAC-seq data for training-regulated features
FEATURE_TO_GENE                                Feature-to-gene map
GENE_UNIVERSES                                 Gene-centric universes
GRAPH_COMPONENTS                               Graph components
GRAPH_PW_ENRICH                                Graph pathway enrichment results
GRAPH_STATES                                   'repfdr' state assignments
GROUP_COLORS                                   Intervention group colors
IMMUNO_ADRNL_DA                                Differential analysis of multiplexed immunoassays
IMMUNO_BAT_DA                                  Differential analysis of multiplexed immunoassays
IMMUNO_COLON_DA                                Differential analysis of multiplexed immunoassays
IMMUNO_CORTEX_DA                               Differential analysis of multiplexed immunoassays
IMMUNO_HEART_DA                                Differential analysis of multiplexed immunoassays
IMMUNO_HIPPOC_DA                               Differential analysis of multiplexed immunoassays
IMMUNO_KIDNEY_DA                               Differential analysis of multiplexed immunoassays
IMMUNO_LIVER_DA                                Differential analysis of multiplexed immunoassays
IMMUNO_LUNG_DA                                 Differential analysis of multiplexed immunoassays
IMMUNO_META                                    Multiplexed immunoassay metadata and QC
IMMUNO_NORM_DATA_FLAT                          Combined immunoassay data used for visualization
IMMUNO_NORM_DATA_NESTED                        Processed immunoassay data used for differential analysis
IMMUNO_OVARY_DA                                Differential analysis of multiplexed immunoassays
IMMUNO_PLASMA_DA                               Differential analysis of multiplexed immunoassays
IMMUNO_SKMGN_DA                                Differential analysis of multiplexed immunoassays
IMMUNO_SKMVL_DA                                Differential analysis of multiplexed immunoassays
IMMUNO_SMLINT_DA                               Differential analysis of multiplexed immunoassays
IMMUNO_SPLEEN_DA                               Differential analysis of multiplexed immunoassays
IMMUNO_TESTES_DA                               Differential analysis of multiplexed immunoassays
IMMUNO_WATSC_DA                                Differential analysis of multiplexed immunoassays
METAB_ADRNL_DA                                 Differential analysis of merged metabolomics datasets
METAB_ADRNL_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METAB_BAT_DA                                   Differential analysis of merged metabolomics datasets
METAB_BAT_DA_METAREG                           Meta-regression of metabolomics differential analysis results
METAB_COLON_DA                                 Differential analysis of merged metabolomics datasets
METAB_COLON_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METAB_CORTEX_DA                                Differential analysis of merged metabolomics datasets
METAB_CORTEX_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_FEATURE_ID_MAP                           Metabolite feature ID map
METAB_HEART_DA                                 Differential analysis of merged metabolomics datasets
METAB_HEART_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METAB_HIPPOC_DA                                Differential analysis of merged metabolomics datasets
METAB_HIPPOC_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_HYPOTH_DA                                Differential analysis of merged metabolomics datasets
METAB_HYPOTH_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_KIDNEY_DA                                Differential analysis of merged metabolomics datasets
METAB_KIDNEY_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_LIVER_DA                                 Differential analysis of merged metabolomics datasets
METAB_LIVER_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METAB_LUNG_DA                                  Differential analysis of merged metabolomics datasets
METAB_LUNG_DA_METAREG                          Meta-regression of metabolomics differential analysis results
METAB_NORM_DATA_FLAT                           Combined metabolomics data used for visualization
METAB_OVARY_DA                                 Differential analysis of merged metabolomics datasets
METAB_OVARY_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METAB_PLASMA_DA                                Differential analysis of merged metabolomics datasets
METAB_PLASMA_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_SAMPLE_DATA_NESTED                       Nested metabolomics data used for differential analysis
METAB_SKMGN_DA                                 Differential analysis of merged metabolomics datasets
METAB_SKMGN_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METAB_SKMVL_DA                                 Differential analysis of merged metabolomics datasets
METAB_SKMVL_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METAB_SMLINT_DA                                Differential analysis of merged metabolomics datasets
METAB_SMLINT_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_SPLEEN_DA                                Differential analysis of merged metabolomics datasets
METAB_SPLEEN_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_TESTES_DA                                Differential analysis of merged metabolomics datasets
METAB_TESTES_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_VENACV_DA                                Differential analysis of merged metabolomics datasets
METAB_VENACV_DA_METAREG                        Meta-regression of metabolomics differential analysis results
METAB_WATSC_DA                                 Differential analysis of merged metabolomics datasets
METAB_WATSC_DA_METAREG                         Meta-regression of metabolomics differential analysis results
METHYL_BAT_NORM_DATA_05FDR                     Normalized DNA methylation data for training-regulated features
METHYL_HEART_NORM_DATA_05FDR                   Normalized DNA methylation data for training-regulated features
METHYL_HIPPOC_NORM_DATA_05FDR                  Normalized DNA methylation data for training-regulated features
METHYL_KIDNEY_NORM_DATA_05FDR                  Normalized DNA methylation data for training-regulated features
METHYL_LIVER_NORM_DATA_05FDR                   Normalized DNA methylation data for training-regulated features
METHYL_LUNG_NORM_DATA_05FDR                    Normalized DNA methylation data for training-regulated features
METHYL_META                                    RRBS metadata and QC
METHYL_SKMGN_NORM_DATA_05FDR                   Normalized DNA methylation data for training-regulated features
METHYL_WATSC_NORM_DATA_05FDR                   Normalized DNA methylation data for training-regulated features
OUTLIERS                                       Sample outliers
PATHWAY_PARENTS                                KEGG and Reactome parent pathways
PHENO                                          Phenotypic data
PHOSPHO_CORTEX_DA                              Differential analysis of proteomics datasets
PHOSPHO_CORTEX_NORM_DATA                       Normalized protein phosphorylation data
PHOSPHO_HEART_DA                               Differential analysis of proteomics datasets
PHOSPHO_HEART_NORM_DATA                        Normalized protein phosphorylation data
PHOSPHO_KIDNEY_DA                              Differential analysis of proteomics datasets
PHOSPHO_KIDNEY_NORM_DATA                       Normalized protein phosphorylation data
PHOSPHO_LIVER_DA                               Differential analysis of proteomics datasets
PHOSPHO_LIVER_NORM_DATA                        Normalized protein phosphorylation data
PHOSPHO_LUNG_DA                                Differential analysis of proteomics datasets
PHOSPHO_LUNG_NORM_DATA                         Normalized protein phosphorylation data
PHOSPHO_META                                   Proteomics sample-level metadata
PHOSPHO_SKMGN_DA                               Differential analysis of proteomics datasets
PHOSPHO_SKMGN_NORM_DATA                        Normalized protein phosphorylation data
PHOSPHO_WATSC_DA                               Differential analysis of proteomics datasets
PHOSPHO_WATSC_NORM_DATA                        Normalized protein phosphorylation data
PROT_CORTEX_DA                                 Differential analysis of proteomics datasets
PROT_CORTEX_NORM_DATA                          Normalized protein expression data
PROT_HEART_DA                                  Differential analysis of proteomics datasets
PROT_HEART_NORM_DATA                           Normalized protein expression data
PROT_KIDNEY_DA                                 Differential analysis of proteomics datasets
PROT_KIDNEY_NORM_DATA                          Normalized protein expression data
PROT_LIVER_DA                                  Differential analysis of proteomics datasets
PROT_LIVER_NORM_DATA                           Normalized protein expression data
PROT_LUNG_DA                                   Differential analysis of proteomics datasets
PROT_LUNG_NORM_DATA                            Normalized protein expression data
PROT_META                                      Proteomics sample-level metadata
PROT_SKMGN_DA                                  Differential analysis of proteomics datasets
PROT_SKMGN_NORM_DATA                           Normalized protein expression data
PROT_WATSC_DA                                  Differential analysis of proteomics datasets
PROT_WATSC_NORM_DATA                           Normalized protein expression data
RAT_TO_HUMAN_GENE                              Rat-to-human gene ortholog map
REPEATED_FEATURES                              Repeated feature info
REPFDR_INPUTS                                  'repfdr' inputs
REPFDR_RES                                     'repfdr' results
SEX_COLORS                                     Sex colors
TISSUE_ABBREV                                  Tissue abbreviations
TISSUE_ABBREV_TO_CODE                          Tissue abbreviation-to-code mapping
TISSUE_CODE_TO_ABBREV                          Tissue code-to-abbreviation mapping
TISSUE_COLORS                                  Tissue colors
TISSUE_ORDER                                   Tissue order
TRAINING_REGULATED_FEATURES                    Training-regulated features
TRNSCRPT_ADRNL_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_ADRNL_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_ADRNL_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_BAT_DA                                Differential analysis of RNA-seq datasets
TRNSCRPT_BAT_NORM_DATA                         Normalized RNA-seq data
TRNSCRPT_BAT_RAW_COUNTS                        RNA-seq raw counts
TRNSCRPT_BLOOD_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_BLOOD_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_BLOOD_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_COLON_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_COLON_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_COLON_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_CORTEX_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_CORTEX_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_CORTEX_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_HEART_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_HEART_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_HEART_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_HIPPOC_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_HIPPOC_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_HIPPOC_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_HYPOTH_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_HYPOTH_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_HYPOTH_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_KIDNEY_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_KIDNEY_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_KIDNEY_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_LIVER_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_LIVER_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_LIVER_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_LUNG_DA                               Differential analysis of RNA-seq datasets
TRNSCRPT_LUNG_NORM_DATA                        Normalized RNA-seq data
TRNSCRPT_LUNG_RAW_COUNTS                       RNA-seq raw counts
TRNSCRPT_META                                  RNA-seq metadata and QC
TRNSCRPT_OVARY_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_OVARY_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_OVARY_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_SKMGN_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_SKMGN_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_SKMGN_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_SKMVL_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_SKMVL_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_SKMVL_RAW_COUNTS                      RNA-seq raw counts
TRNSCRPT_SMLINT_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_SMLINT_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_SMLINT_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_SPLEEN_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_SPLEEN_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_SPLEEN_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_TESTES_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_TESTES_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_TESTES_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_VENACV_DA                             Differential analysis of RNA-seq datasets
TRNSCRPT_VENACV_NORM_DATA                      Normalized RNA-seq data
TRNSCRPT_VENACV_RAW_COUNTS                     RNA-seq raw counts
TRNSCRPT_WATSC_DA                              Differential analysis of RNA-seq datasets
TRNSCRPT_WATSC_NORM_DATA                       Normalized RNA-seq data
TRNSCRPT_WATSC_RAW_COUNTS                      RNA-seq raw counts
UBIQ_HEART_DA                                  Differential analysis of proteomics datasets
UBIQ_HEART_NORM_DATA                           Normalized protein ubiquitynation data
UBIQ_LIVER_DA                                  Differential analysis of proteomics datasets
UBIQ_LIVER_NORM_DATA                           Normalized protein ubiquitynation data
UBIQ_META                                      Proteomics sample-level metadata</code></pre>
</div>
<div class="section level2">
<h2 id="access-epigenomics-data-through-google-cloud-storage">Access epigenomics data through Google Cloud Storage<a class="anchor" aria-label="anchor" href="#access-epigenomics-data-through-google-cloud-storage"></a>
</h2>
<p>Due to file size, only normalized sample-level data and differential analysis results corresponding to <strong>training-regulated features</strong> (5% IHW FDR) are contained in this package for chromatin accessibility (ATAC) and DNA methylation (METHYL). The full sets of epigenetic results may be downloaded through the following public URLs. Note that clicking on a URL automatically starts a download.</p>
<table class="table">
<colgroup>
<col width="20%">
<col width="20%">
<col width="20%">
<col width="20%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th>Type</th>
<th>Assay</th>
<th>Tissue</th>
<th>Object</th>
<th>URL</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>BAT</td>
<td>ATAC_BAT_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_BAT_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_BAT_DA.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>HEART</td>
<td>ATAC_HEART_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HEART_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HEART_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>HIPPOC</td>
<td>ATAC_HIPPOC_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HIPPOC_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HIPPOC_DA.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>KIDNEY</td>
<td>ATAC_KIDNEY_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_KIDNEY_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_KIDNEY_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>LIVER</td>
<td>ATAC_LIVER_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LIVER_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LIVER_DA.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>LUNG</td>
<td>ATAC_LUNG_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LUNG_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LUNG_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>SKM-GN</td>
<td>ATAC_SKMGN_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_SKMGN_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_SKMGN_DA.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>ATAC</td>
<td>WAT-SC</td>
<td>ATAC_WATSC_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_WATSC_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_WATSC_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>BAT</td>
<td>ATAC_BAT_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_BAT_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_BAT_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>HEART</td>
<td>ATAC_HEART_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HEART_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HEART_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>HIPPOC</td>
<td>ATAC_HIPPOC_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HIPPOC_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HIPPOC_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>KIDNEY</td>
<td>ATAC_KIDNEY_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_KIDNEY_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_KIDNEY_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>LIVER</td>
<td>ATAC_LIVER_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LIVER_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LIVER_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>LUNG</td>
<td>ATAC_LUNG_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LUNG_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LUNG_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>SKM-GN</td>
<td>ATAC_SKMGN_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_SKMGN_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_SKMGN_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>ATAC</td>
<td>WAT-SC</td>
<td>ATAC_WATSC_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_WATSC_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_WATSC_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>BAT</td>
<td>ATAC_BAT_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_BAT_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_BAT_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>HEART</td>
<td>ATAC_HEART_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HEART_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HEART_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>HIPPOC</td>
<td>ATAC_HIPPOC_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HIPPOC_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_HIPPOC_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>KIDNEY</td>
<td>ATAC_KIDNEY_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_KIDNEY_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_KIDNEY_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>LIVER</td>
<td>ATAC_LIVER_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LIVER_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LIVER_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>LUNG</td>
<td>ATAC_LUNG_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LUNG_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_LUNG_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>SKM-GN</td>
<td>ATAC_SKMGN_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_SKMGN_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_SKMGN_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>ATAC</td>
<td>WAT-SC</td>
<td>ATAC_WATSC_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_WATSC_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_WATSC_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Feature annotation</td>
<td>ATAC</td>
<td>all</td>
<td>ATAC_FEATURE_ANNOT</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_FEATURE_ANNOT.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/ATAC_FEATURE_ANNOT.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>BAT</td>
<td>METHYL_BAT_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>HEART</td>
<td>METHYL_HEART_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_DA.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>HIPPOC</td>
<td>METHYL_HIPPOC_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>KIDNEY</td>
<td>METHYL_KIDNEY_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_DA.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>LIVER</td>
<td>METHYL_LIVER_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>LUNG</td>
<td>METHYL_LUNG_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_DA.rda</a></td>
</tr>
<tr class="even">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>SKM-GN</td>
<td>METHYL_SKMGN_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_DA.rda</a></td>
</tr>
<tr class="odd">
<td>Differential analysis results</td>
<td>METHYL</td>
<td>WAT-SC</td>
<td>METHYL_WATSC_DA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_DA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_DA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>BAT</td>
<td>METHYL_BAT_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>HEART</td>
<td>METHYL_HEART_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>HIPPOC</td>
<td>METHYL_HIPPOC_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>KIDNEY</td>
<td>METHYL_KIDNEY_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>LIVER</td>
<td>METHYL_LIVER_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>LUNG</td>
<td>METHYL_LUNG_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>SKM-GN</td>
<td>METHYL_SKMGN_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_NORM_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Normalized sample-level data</td>
<td>METHYL</td>
<td>WAT-SC</td>
<td>METHYL_WATSC_NORM_DATA</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_NORM_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_NORM_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Raw data</td>
<td>METHYL</td>
<td>BAT</td>
<td>BAT_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_RAW_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Raw data</td>
<td>METHYL</td>
<td>HEART</td>
<td>HEART_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_RAW_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Raw data</td>
<td>METHYL</td>
<td>HIPPOC</td>
<td>HIPPOC_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_RAW_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Raw data</td>
<td>METHYL</td>
<td>KIDNEY</td>
<td>KIDNEY_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_RAW_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Raw data</td>
<td>METHYL</td>
<td>LIVER</td>
<td>LIVER_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_RAW_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Raw data</td>
<td>METHYL</td>
<td>LUNG</td>
<td>LUNG_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_RAW_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Raw data</td>
<td>METHYL</td>
<td>SKM-GN</td>
<td>SKMGN_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_RAW_DATA.rda</a></td>
</tr>
<tr class="odd">
<td>Raw data</td>
<td>METHYL</td>
<td>WAT-SC</td>
<td>WATSC_raw</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_RAW_DATA.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_RAW_DATA.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>BAT</td>
<td>METHYL_BAT_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>HEART</td>
<td>METHYL_HEART_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>HIPPOC</td>
<td>METHYL_HIPPOC_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>KIDNEY</td>
<td>METHYL_KIDNEY_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>LIVER</td>
<td>METHYL_LIVER_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>LUNG</td>
<td>METHYL_LUNG_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>SKM-GN</td>
<td>METHYL_SKMGN_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_RAW_COUNTS.rda</a></td>
</tr>
<tr class="odd">
<td>Raw sample-level counts</td>
<td>METHYL</td>
<td>WAT-SC</td>
<td>METHYL_WATSC_RAW_COUNTS</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_RAW_COUNTS.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_RAW_COUNTS.rda</a></td>
</tr>
<tr class="even">
<td>Feature annotation</td>
<td>METHYL</td>
<td>all</td>
<td>METHYL_FEATURE_ANNOT</td>
<td><a href="https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_FEATURE_ANNOT.rda" class="external-link uri">https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_FEATURE_ANNOT.rda</a></td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h2>
<p>For questions, bug reporting, and data requests for this package, please <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/issues" class="external-link">submit a new issue</a> and include as many details as possible.</p>
<p>If the concern is related to functions provided in the <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo" class="external-link">MotrpacRatTraining6mo</a> package, please submit an issue <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/issues" class="external-link">here</a> instead.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>MoTrPAC is supported by the National Institutes of Health (NIH) Common Fund through cooperative agreements managed by the National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), and National Institute on Aging (NIA).</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing MotrpacRatTraining6moData</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Nicole Gay <br><small class="roles"> Maintainer, author </small>  </li>
<li>Pierre Jean Beltran <br><small class="roles"> Author </small>  </li>
<li>David Amar <br><small class="roles"> Author </small>  </li>
<li>David Jimenez-Morales <br><small class="roles"> Author </small>  </li>
<li>MoTrPAC Study Group <br><small class="roles"> Author, data contributor </small>  </li>
<li>NIH Common Fund <br><small class="roles"> Funder </small>  </li>
<li>National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK) <br><small class="roles"> Funder </small>  </li>
<li>National Institute of Arthritis and Musculoskeletal Diseases (NIAMS) <br><small class="roles"> Funder </small>  </li>
<li>National Institute on Aging (NIA) <br><small class="roles"> Funder </small>  </li>
<li>National Institute of Biomedical Imaging and Bioengineering <br><small class="roles"> Funder </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
