<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MotrpacRatTraining6moData">
<title>Tutorial for MotrpacRatTraining6moData R package • MotrpacRatTraining6moData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial for MotrpacRatTraining6moData R package">
<meta property="og:description" content="MotrpacRatTraining6moData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6moData.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Tutorial for MotrpacRatTraining6moData R package</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/blob/HEAD/vignettes/MotrpacRatTraining6moData.Rmd" class="external-link"><code>vignettes/MotrpacRatTraining6moData.Rmd</code></a></small>
      <div class="d-none name"><code>MotrpacRatTraining6moData.Rmd</code></div>
    </div>

    
    
<style>
div.blue {
  background-color: #e6f0ff; 
  border-radius: 10px; 
  padding-top: 20px; 
  padding-right: 20px; 
  padding-left: 20px; 
  padding-bottom: 10px; 
  margin-bottom: 20px; 
}
div.yellow {
  background-color: #fff5e6; 
  border-radius: 10px; 
  padding-top: 20px; 
  padding-right: 20px; 
  padding-left: 20px; 
  padding-bottom: 10px; 
  margin-bottom: 20px; 
}
#TOC {
  top: 1%;
  opacity: 0.5;
}
#TOC:hover {
  opacity: 1;
}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="yellow">
<p>This vignette is under construction. If something is unclear or
obviously missing, please submit an issue <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/issues" class="external-link">here</a>.</p>
</div>
<div class="section level3">
<h3 id="about-this-package">About this package<a class="anchor" aria-label="anchor" href="#about-this-package"></a>
</h3>
<p>This package provides convenient access to the processed data and
downstream analysis results presented in the main paper for the first
large-scale multi-omic multi-tissue endurance exercise training study
conducted in young adult rats by the Molecular Transducers of Physical
Activity Consortium (MoTrPAC). Find the <a href="https://www.biorxiv.org/content/10.1101/2022.09.21.508770v2" class="external-link">preprint
on bioRxiv</a>. <strong>We highly recommend skimming the preprint before
using this package as it provides important context and much greater
detail than we can provide here.</strong></p>
<div class="blue">
<p>While the data in this package can be used by themselves, as
demonstrated in this vignette, the <a href="https://motrpac.github.io/MotrpacRatTraining6mo/" class="external-link">MotrpacRatTraining6mo</a>
R package relies heavily on this package and provides many functions to
help retrieve and explore the data. See the <a href="https://motrpac.github.io/MotrpacRatTraining6mo/articles/MotrpacRatTraining6mo.html" class="external-link">MotrpacRatTraining6mo
vignette</a> for examples.</p>
</div>
</div>
<div class="section level3">
<h3 id="about-motrpac">About MoTrPAC<a class="anchor" aria-label="anchor" href="#about-motrpac"></a>
</h3>
<p>MoTrPAC is a national research consortium designed to discover and
perform preliminary characterization of the range of molecular
transducers (the “molecular map”) that underlie the effects of physical
activity in humans. The program’s goal is to study the molecular changes
that occur during and after exercise and ultimately to advance the
understanding of how physical activity improves and preserves health.
The six-year program is the largest targeted NIH investment of funds
into the mechanisms of how physical activity improves health and
prevents disease. See <a href="https://www.motrpac.org/" class="external-link">motrpac.org</a>
and <a href="https://motrpac-data.org/" class="external-link">motrpac-data.org</a> for more
details.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Install this package with <code>devtools</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MoTrPAC/MotrpacRatTraining6moData"</span><span class="op">)</span></span></code></pre></div>
<p>The output for a successful installation looks something like this.
Note that the <code>*** moving datasets to lazyload DB</code> step takes
the longest (~5 minutes):</p>
<pre><code>Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
✓  checking for file ‘.../MoTrPAC-MotrpacRatTraining6moData-1c6478a/DESCRIPTION’ ...
─  preparing ‘MotrpacRatTraining6moData’:
✓  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘MotrpacRatTraining6moData_1.3.2.tar.gz’ (1.3s)
   
* installing *source* package ‘MotrpacRatTraining6moData’ ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (MotrpacRatTraining6moData)</code></pre>
<div class="section level4">
<h4 id="troubleshooting">Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h4>
<p>If you get this error:</p>
<pre><code>Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD' failed</code></pre>
<p>..try extending the timeout:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="fl">1e5</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MoTrPAC/MotrpacRatTraining6moData"</span><span class="op">)</span></span></code></pre></div>
<p>If you get this error after extending the <code>timeout</code>:</p>
<pre><code>Downloading GitHub repo MoTrPAC/MotrpacRatTraining6moData@HEAD
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD' failed
Error in `action()`:
! `class` is absent but must be supplied.
Run `rlang::last_error()` to see where the error occurred.</code></pre>
<p>…this seems to be an intermittent issue seen only on Mac, not Linux
or Windows. This was resolved by <a href="https://cran.rstudio.com/" class="external-link">installing the newest version of
R</a>.</p>
</div>
<div class="section level4">
<h4 id="last-resort">Last resort<a class="anchor" aria-label="anchor" href="#last-resort"></a>
</h4>
<p>If you can’t get
<code>devtools::install_github("MoTrPAC/MotrpacRatTraining6moData")</code>
to work, try this:</p>
<ol style="list-style-type: decimal">
<li><p>Go to <a href="https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD" class="external-link uri">https://api.github.com/repos/MoTrPAC/MotrpacRatTraining6moData/tarball/HEAD</a>,
which will automatically start downloading this repository in a
tarball</p></li>
<li>
<p>Install the package from source:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"~/Downloads/MoTrPAC-MotrpacRatTraining6moData-0729e2e.tar.gz"</span>, </span>
<span>  repos <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>  type <span class="op">=</span> <span class="st">"source"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moData/">MotrpacRatTraining6moData</a></span><span class="op">)</span></span></code></pre></div>
</li>
</ol>
<p>Once we install this package, we can load the library.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moData/">MotrpacRatTraining6moData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span> <span class="co"># for plots in this tutorial </span></span></code></pre></div>
<div>
<p><strong>Tip:</strong> To learn more about any data object, use
<code>?</code> to retrieve the documentation, e.g.,
<code><a href="../reference/METAB_FEATURE_ID_MAP.html">?METAB_FEATURE_ID_MAP</a></code>. Note that
<code>MotrpacRatTraining6moData</code> must be installed and loaded with
<code><a href="https://rdrr.io/r/base/library.html" class="external-link">library()</a></code> for this to work.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="study-design">Study design<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h2>
<p>Details of the experimental design can be found in the <a href="https://www.biorxiv.org/content/biorxiv/early/2022/10/05/2022.09.21.508770/DC1/embed/media-1.pdf?download=true" class="external-link">supplementary
methods of the bioRxiv preprint</a>. Briefly, 6-month-old young adult
rats were subjected to progressive endurance exercise training for 1, 2,
4, or 8 weeks, with tissues collected 48 hours after the last training
bout. Sex-matched sedentary, untrained rats were used as controls. Whole
blood, plasma, and 18 solid tissues were analyzed using genomics,
proteomics, metabolomics, and protein immunoassay technologies, with
most assays performed in a subset of these tissues. Depending on the
assay, between 3 and 6 replicates per sex per time point were
analyzed.</p>
</div>
<div class="section level2">
<h2 id="tissue-and-assay-abbreviations">Tissue and assay abbreviations<a class="anchor" aria-label="anchor" href="#tissue-and-assay-abbreviations"></a>
</h2>
<p>It is important to be aware of the tissue and assay abbreviations
because they are used to name many data objects. The vectors of
abbreviations are also available in <code>TISSUE_ABBREV</code> and
<code>ASSAY_ABBREV</code>.</p>
<div class="section level3">
<h3 id="tissues">Tissues<a class="anchor" aria-label="anchor" href="#tissues"></a>
</h3>
<ul>
<li>
<strong>ADRNL:</strong> adrenal gland<br>
</li>
<li>
<strong>BAT:</strong> brown adipose tissue<br>
</li>
<li>
<strong>BLOOD:</strong> whole blood<br>
</li>
<li>
<strong>COLON:</strong> colon<br>
</li>
<li>
<strong>CORTEX:</strong> cerebral cortex<br>
</li>
<li>
<strong>HEART:</strong> heart<br>
</li>
<li>
<strong>HIPPOC:</strong> hippocampus<br>
</li>
<li>
<strong>HYPOTH:</strong> hypothalamus<br>
</li>
<li>
<strong>KIDNEY:</strong> kidney<br>
</li>
<li>
<strong>LIVER:</strong> liver<br>
</li>
<li>
<strong>LUNG:</strong> lung<br>
</li>
<li>
<strong>OVARY:</strong> ovaries (female gonads)<br>
</li>
<li>
<strong>PLASMA:</strong> plasma from blood</li>
<li>
<strong>SKM-GN:</strong> gastrocnemius (leg skeletal muscle)<br>
</li>
<li>
<strong>SKM-VL:</strong> vastus lateralis (leg skeletal muscle)</li>
<li>
<strong>SMLINT:</strong> small intestine<br>
</li>
<li>
<strong>SPLEEN:</strong> spleen<br>
</li>
<li>
<strong>TESTES:</strong> testes (male gonads)<br>
</li>
<li>
<strong>VENACV:</strong> vena cava<br>
</li>
<li>
<strong>WAT-SC:</strong> subcutaneous white adipose tissue</li>
</ul>
</div>
<div class="section level3">
<h3 id="assaysomes">Assays/omes<a class="anchor" aria-label="anchor" href="#assaysomes"></a>
</h3>
<ul>
<li>
<strong>ACETYL:</strong> acetylproteomics; protein site
acetylation<br>
</li>
<li>
<strong>ATAC:</strong> chromatin accessibility, ATAC-seq data<br>
</li>
<li>
<strong>IMMUNO:</strong> multiplexed immunoassays (cytokines and
hormones)<br>
</li>
<li>
<strong>METAB:</strong> metabolomics and lipidomics<br>
</li>
<li>
<strong>METHYL:</strong> DNA methylation, RRBS data<br>
</li>
<li>
<strong>PHOSPHO:</strong> phosphoproteomics; protein site
phosphorylation<br>
</li>
<li>
<strong>PROT:</strong> global proteomics; protein abundance<br>
</li>
<li>
<strong>TRNSCRPT:</strong> transcriptomics, RNA-Seq data<br>
</li>
<li>
<strong>UBIQ:</strong> ubiquitylome; protein site
ubiquitination</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="summary-of-data">Summary of data<a class="anchor" aria-label="anchor" href="#summary-of-data"></a>
</h2>
<p>Here is a brief summary of the kinds of data included in this
package:</p>
<ul>
<li>Assay, tissue, sex, and training group abbreviations, codes, colors,
and order used in plots</li>
<li>Phenotypic data, <code>PHENO</code><br>
</li>
<li>Mapping between various feature identifiers, i.e.,
<code>FEATURE_TO_GENE</code>, <code>RAT_TO_HUMAN_GENE</code><br>
</li>
<li>Ome-specific feature annotation, i.e.,
<code>METAB_FEATURE_ID_MAP</code>, <code>METHYL_FEATURE_ANNOT</code>
(GCP only), <code>ATAC_FEATURE_ANNOT</code> (GCP only)</li>
<li>Ome-specific sample-level metadata, i.e.,
<code>TRNSCRPT_META</code>, <code>ATAC_META</code>,
<code>METHYL_META</code>, <code>IMMUNO_META</code>,
<code>PHOSPHO_META</code>, <code>PROT_META</code>,
<code>ACETYL_META</code>, <code>UBIQ_META</code><br>
</li>
<li>Raw counts for RNA-Seq (TRNSCRPT), ATAC-Seq (ATAC), and RRBS
(METHYL) data, e.g., <code>TRNSCRPT_LIVER_RAW_COUNTS</code>. Note that
epigenetic data (ATAC and METHYL) must be downloaded from Google Cloud
Storage. See more details <a href="https://motrpac.github.io/MotrpacRatTraining6moData/index.html#access-epigenomics-data-through-google-cloud-storage">here</a>.<br>
</li>
<li>Normalized sample-level data, e.g.,
<code>TRNSCRPT_SKMGN_NORM_DATA</code><br>
</li>
<li>Differential analysis results, e.g., <code>HEART_PROT_DA</code>
</li>
<li>Sample outliers excluded from differential analysis,
<code>OUTLIERS</code><br>
</li>
<li>Table of training-regulated features at 5% FDR,
<code>TRAINING_REGULATED_FEATURES</code><br>
</li>
<li>Bayesian graphical analysis inputs and results</li>
<li>Pathway enrichment of main graphical clusters,
<code>GRAPH_PW_ENRICH</code>
</li>
</ul>
<p>A list of all the available data objects and a brief description are
available <a href="https://motrpac.github.io/MotrpacRatTraining6moData/index.html#explore-available-data-objects">here</a>.</p>
</div>
<div class="section level2">
<h2 id="principal-component-analysis">Principal component analysis<a class="anchor" aria-label="anchor" href="#principal-component-analysis"></a>
</h2>
<p>Here we show how to combine the normalized RNA-seq data from the
gastrocnemius (leg skeletal muscle) with the phenotypic data to perform
principal component analysis (PCA). We also take advantage of the
training group colors.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load data </span></span>
<span><span class="co"># While this is not necessary, we want to make it clear what objects are available in the package.</span></span>
<span><span class="co"># Note you can use data directly without loading with `data()`, e.g., `head(PHENO)`</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PHENO</span><span class="op">)</span> <span class="co"># phenotypic data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">TRNSCRPT_SKMGN_NORM_DATA</span><span class="op">)</span> <span class="co"># normalized gastrocnemius RNA-seq data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GROUP_COLORS</span><span class="op">)</span> <span class="co"># hex colors for training groups </span></span>
<span></span>
<span><span class="co"># The first four columns in the sample-level data specify row meta-data; next columns are samples</span></span>
<span><span class="co"># Note the "feature" column is only non-NA for training-regulated features at 5% FDR</span></span>
<span><span class="va">TRNSCRPT_SKMGN_NORM_DATA</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span><span class="co">#&gt;   feature         feature_ID tissue    assay 90560015512 90581015512</span></span>
<span><span class="co">#&gt; 1    &lt;NA&gt; ENSRNOG00000000008 SKM-GN TRNSCRPT     0.04044    -0.09760</span></span>
<span><span class="co">#&gt; 2    &lt;NA&gt; ENSRNOG00000000012 SKM-GN TRNSCRPT     2.61487     2.78841</span></span>
<span><span class="co">#&gt; 3    &lt;NA&gt; ENSRNOG00000000021 SKM-GN TRNSCRPT     2.17049     1.70091</span></span>
<span><span class="va">sample_ids</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">TRNSCRPT_SKMGN_NORM_DATA</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Perform PCA</span></span>
<span><span class="va">skmgn_pca</span> <span class="op">=</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">TRNSCRPT_SKMGN_NORM_DATA</span><span class="op">[</span>,<span class="va">sample_ids</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make a data frame with some phenotypic data and the first 3 PCs</span></span>
<span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  group <span class="op">=</span> <span class="va">PHENO</span><span class="op">[</span><span class="va">sample_ids</span>,<span class="st">"group"</span><span class="op">]</span>,</span>
<span>  sex <span class="op">=</span> <span class="va">PHENO</span><span class="op">[</span><span class="va">sample_ids</span>,<span class="st">"sex"</span><span class="op">]</span>,</span>
<span>  <span class="va">skmgn_pca</span><span class="op">$</span><span class="va">x</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="co"># take the first principal components</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the first two PCs</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">PC1</span>, y<span class="op">=</span><span class="va">PC2</span>, colour<span class="op">=</span><span class="va">group</span>, shape<span class="op">=</span><span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> </span></code></pre></div>
<p><img src="MotrpacRatTraining6moData_files/figure-html/unnamed-chunk-2-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Use GROUP_COLORS and make it prettier </span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">PC1</span>, y<span class="op">=</span><span class="va">PC2</span>, fill<span class="op">=</span><span class="va">group</span>, shape<span class="op">=</span><span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span>, colour<span class="op">=</span><span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="va">GROUP_COLORS</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_shape_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>male<span class="op">=</span><span class="fl">21</span>, female<span class="op">=</span><span class="fl">24</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>fill<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>shape<span class="op">=</span><span class="fl">21</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title<span class="op">=</span><span class="st">"PCA of Gastrocnemius RNA-seq data"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6moData_files/figure-html/unnamed-chunk-2-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="compare-rna-and-protein-data">Compare RNA and protein data<a class="anchor" aria-label="anchor" href="#compare-rna-and-protein-data"></a>
</h2>
<p>Here we take the heart RNA-seq and global proteomics data, match
their sample IDs, and map the features to gene IDs.</p>
<div class="section level3">
<h3 id="data-processing">Data processing<a class="anchor" aria-label="anchor" href="#data-processing"></a>
</h3>
<p>First, load the normalized data from each ome.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rnaseq_d</span> <span class="op">=</span> <span class="va">TRNSCRPT_HEART_NORM_DATA</span></span>
<span><span class="va">prot_d</span> <span class="op">=</span> <span class="va">PROT_HEART_NORM_DATA</span></span></code></pre></div>
<p>By default, the column names of most sample-level data are vial
labels, which are sample-specific identifiers. Different samples from
the sample animal were processed for different assays. Therefore, in
order to map between samples from the same animal across assays, we need
to map sample IDs to participant IDs (PIDs) or biospecimen IDs (BIDs).
Here, we map vial labels (sample IDs) to BIDs because BIDs are
conveniently a substring of the vial labels. <code>PHENO</code> can also
be used to map between vial labels (<code>viallabel</code>), PID
(<code>pid</code>), and BID (<code>bid</code>).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Match sample ids</span></span>
<span><span class="co">#' A helper function for getting a named vector that maps biospecimen ids to sample ids</span></span>
<span><span class="co">#' The biospecimen id of each sample is the substring of the sample id.</span></span>
<span><span class="va">map_sample_to_bid</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substring</a></span><span class="op">(</span>first<span class="op">=</span><span class="fl">1</span>, last<span class="op">=</span><span class="fl">5</span>, <span class="va">x</span><span class="op">)</span></span>
<span>  <span class="va">x</span></span>
<span><span class="op">}</span></span>
<span><span class="va">rnaseq_bid</span> <span class="op">=</span> <span class="fu">map_sample_to_bid</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rnaseq_d</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">prot_bid</span> <span class="op">=</span> <span class="fu">map_sample_to_bid</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">prot_d</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">shared_bids</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">rnaseq_bid</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prot_bid</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now we used the <code>FEATURE_TO_GENE</code> map to map RNA-seq and
proteomics feature IDs to gene symbols. <code>FEATURE_TO_GENE</code> is
very large (&gt;4M rows), so we start by subsetting it to the features
in the current analysis.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">FEATURE_TO_GENE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 4044034       9</span></span>
<span><span class="va">f2g</span> <span class="op">=</span> <span class="va">FEATURE_TO_GENE</span><span class="op">[</span></span>
<span>  <span class="va">FEATURE_TO_GENE</span><span class="op">$</span><span class="va">feature_ID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">rnaseq_d</span><span class="op">$</span><span class="va">feature_ID</span>,<span class="va">prot_d</span><span class="op">$</span><span class="va">feature_ID</span><span class="op">)</span>,</span>
<span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">f2g</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 24591     9</span></span></code></pre></div>
<p>Use the sample-to-BID mapping and the feature-to-gene mapping to
subset the RNA-seq and proteomics data to overlapping BIDs and genes.
For cases where multiple feature IDs map to the same gene, take the
average value per BID.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get new RNA-seq and proteomics data frames</span></span>
<span><span class="co"># here we compute the gene-wise average profiles, while limiting the</span></span>
<span><span class="co"># data to the shared biospecimen ids</span></span>
<span><span class="va">rnaseq_x</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">rnaseq_d</span>, <span class="va">f2g</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature_ID"</span>,<span class="st">"gene_symbol"</span><span class="op">)</span><span class="op">]</span>, by<span class="op">=</span><span class="st">"feature_ID"</span><span class="op">)</span></span>
<span><span class="va">rnaseq_x</span> <span class="op">=</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span></span>
<span>  <span class="va">rnaseq_x</span><span class="op">[</span>,<span class="va">rnaseq_bid</span><span class="op">[</span><span class="va">shared_bids</span><span class="op">]</span><span class="op">]</span>, <span class="co"># get the shared samples</span></span>
<span>  by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">rnaseq_x</span><span class="op">$</span><span class="va">gene_symbol</span><span class="op">)</span>, <span class="va">mean</span>, na.rm<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Same for proteomics</span></span>
<span><span class="va">prot_x</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">prot_d</span>,<span class="va">f2g</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature_ID"</span>,<span class="st">"gene_symbol"</span><span class="op">)</span><span class="op">]</span>,by<span class="op">=</span><span class="st">"feature_ID"</span><span class="op">)</span></span>
<span><span class="va">prot_x</span> <span class="op">=</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span></span>
<span>  <span class="va">prot_x</span><span class="op">[</span>,<span class="va">prot_bid</span><span class="op">[</span><span class="va">shared_bids</span><span class="op">]</span><span class="op">]</span>, <span class="co"># get the shared samples</span></span>
<span>  by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">prot_x</span><span class="op">$</span><span class="va">gene_symbol</span><span class="op">)</span>,<span class="va">mean</span>,na.rm<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove proteomics genes with missing values</span></span>
<span><span class="va">prot_x</span> <span class="op">=</span> <span class="va">prot_x</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">prot_x</span><span class="op">)</span>,<span class="fl">1</span>,<span class="va">any</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># Subset the data to the shared genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rnaseq_x</span><span class="op">)</span> <span class="op">=</span> <span class="va">rnaseq_x</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">rnaseq_x</span> <span class="op">=</span> <span class="va">rnaseq_x</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_x</span><span class="op">)</span> <span class="op">=</span> <span class="va">prot_x</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">prot_x</span> <span class="op">=</span> <span class="va">prot_x</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">shared_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rnaseq_x</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_x</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">prot_x</span> <span class="op">=</span> <span class="va">prot_x</span><span class="op">[</span><span class="va">shared_genes</span>,<span class="op">]</span></span>
<span><span class="va">rnaseq_x</span> <span class="op">=</span> <span class="va">rnaseq_x</span><span class="op">[</span><span class="va">shared_genes</span>,<span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-abundance-analysis">Differential abundance analysis<a class="anchor" aria-label="anchor" href="#differential-abundance-analysis"></a>
</h3>
<p>Now we can perform a simple differential analysis between the males
trained for 8 weeks and the sedentary control males to identify genes
that are regulated by training. We do this separately for RNA-seq and
proteomics data. We use t-tests here for simplicity, but a more
appropriate way to compute differential abundance is to use tools like
<code>limma</code> and <code>DESeq2</code>. Fortunately, these more
robust results are also available in this package!</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform simple t-tests</span></span>
<span><span class="va">ttest_wrapper</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">groups</span>, <span class="va">group1</span>, <span class="va">group2</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">x1</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">groups</span><span class="op">==</span><span class="va">group1</span><span class="op">]</span></span>
<span>  <span class="va">x2</span> <span class="op">=</span> <span class="va">x</span><span class="op">[</span><span class="va">groups</span><span class="op">==</span><span class="va">group2</span><span class="op">]</span></span>
<span>  <span class="va">tres</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="va">x1</span>,<span class="va">x2</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"logFC"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">tres</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">-</span><span class="va">tres</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="st">"tscore"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">tres</span><span class="op">$</span><span class="va">statistic</span><span class="op">)</span>,</span>
<span>    <span class="st">"p_value"</span> <span class="op">=</span> <span class="va">tres</span><span class="op">$</span><span class="va">p.value</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Compute t-tests for week 8 in males</span></span>
<span><span class="va">training_groups</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span></span>
<span>  <span class="va">PHENO</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rnaseq_x</span><span class="op">)</span>,<span class="st">"sex"</span><span class="op">]</span>,</span>
<span>  <span class="va">PHENO</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rnaseq_x</span><span class="op">)</span>,<span class="st">"group"</span><span class="op">]</span>, sep<span class="op">=</span><span class="st">","</span></span>
<span><span class="op">)</span></span>
<span><span class="va">male_8w_rnaseq_ttests</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">rnaseq_x</span>,</span>
<span>                              <span class="fl">1</span>,</span>
<span>                              <span class="va">ttest_wrapper</span>,</span>
<span>                              groups<span class="op">=</span><span class="va">training_groups</span>,</span>
<span>                              group1<span class="op">=</span><span class="st">"male,8w"</span>,</span>
<span>                              group2<span class="op">=</span><span class="st">"male,control"</span><span class="op">)</span></span>
<span><span class="va">male_8w_prot_ttests</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">prot_x</span>,</span>
<span>                            <span class="fl">1</span>,</span>
<span>                            <span class="va">ttest_wrapper</span>,</span>
<span>                            groups<span class="op">=</span><span class="va">training_groups</span>,</span>
<span>                            group1<span class="op">=</span><span class="st">"male,8w"</span>,</span>
<span>                            group2<span class="op">=</span><span class="st">"male,control"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># As we observed in the paper, the differential analysis results are mildly correlated:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">male_8w_rnaseq_ttests</span><span class="op">[</span><span class="st">"tscore"</span>,<span class="op">]</span>, <span class="va">male_8w_prot_ttests</span><span class="op">[</span><span class="st">"tscore"</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.1920838</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare-to-published-results">Compare to published results<a class="anchor" aria-label="anchor" href="#compare-to-published-results"></a>
</h3>
<p>Now we compare the results from the simple t-test above to the
differential analysis results from DESeq2 (for RNA-seq) and limma (for
proteomics).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rnaseq_DA</span> <span class="op">=</span> <span class="va">TRNSCRPT_HEART_DA</span></span>
<span><span class="va">prot_DA</span> <span class="op">=</span> <span class="va">PROT_HEART_DA</span></span>
<span></span>
<span><span class="co"># subset the results to 8w, males</span></span>
<span><span class="va">rnaseq_DA</span> <span class="op">=</span> <span class="va">rnaseq_DA</span><span class="op">[</span><span class="va">rnaseq_DA</span><span class="op">$</span><span class="va">sex</span> <span class="op">==</span> <span class="st">"male"</span> <span class="op">&amp;</span> <span class="va">rnaseq_DA</span><span class="op">$</span><span class="va">comparison_group</span><span class="op">==</span><span class="st">"8w"</span>,<span class="op">]</span></span>
<span><span class="va">prot_DA</span> <span class="op">=</span> <span class="va">prot_DA</span><span class="op">[</span><span class="va">prot_DA</span><span class="op">$</span><span class="va">sex</span> <span class="op">==</span> <span class="st">"male"</span> <span class="op">&amp;</span> <span class="va">prot_DA</span><span class="op">$</span><span class="va">comparison_group</span><span class="op">==</span><span class="st">"8w"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># Add gene symbols</span></span>
<span><span class="va">rnaseq_DA</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">rnaseq_DA</span>, <span class="va">f2g</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature_ID"</span>,<span class="st">"gene_symbol"</span><span class="op">)</span><span class="op">]</span>, by<span class="op">=</span><span class="st">"feature_ID"</span><span class="op">)</span></span>
<span><span class="va">prot_DA</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">prot_DA</span>, <span class="va">f2g</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature_ID"</span>,<span class="st">"gene_symbol"</span><span class="op">)</span><span class="op">]</span>, by<span class="op">=</span><span class="st">"feature_ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get the best t-test results per gene</span></span>
<span><span class="va">get_best_zscore</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">zs</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">zs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">zs</span><span class="op">)</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">zs</span><span class="op">)</span>,na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">rnaseq_t</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">rnaseq_DA</span><span class="op">$</span><span class="va">zscore</span>,</span>
<span>                  <span class="va">rnaseq_DA</span><span class="op">$</span><span class="va">gene_symbol</span>,</span>
<span>                  <span class="va">get_best_zscore</span><span class="op">)</span></span>
<span><span class="va">prot_t</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">prot_DA</span><span class="op">$</span><span class="va">tscore</span>,</span>
<span>                <span class="va">prot_DA</span><span class="op">$</span><span class="va">gene_symbol</span>,</span>
<span>                <span class="va">get_best_zscore</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Again, we see mild but non-zero correlation</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">rnaseq_t</span><span class="op">[</span><span class="va">shared_genes</span><span class="op">]</span>,<span class="va">prot_t</span><span class="op">[</span><span class="va">shared_genes</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.2126466</span></span>
<span></span>
<span><span class="co"># Here we show an almost perfect correlation between the pre-computed </span></span>
<span><span class="co"># RNA-seq DESeq2 results and the simple t-tests performed above</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">rnaseq_t</span><span class="op">[</span><span class="va">shared_genes</span><span class="op">]</span>,</span>
<span>     <span class="va">male_8w_rnaseq_ttests</span><span class="op">[</span><span class="st">"tscore"</span>,<span class="va">shared_genes</span><span class="op">]</span>,</span>
<span>     main <span class="op">=</span> <span class="st">"Transcriptomics DA, males at week 8"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"DESeq2 t-score"</span>,ylab <span class="op">=</span> <span class="st">"Simple t-test statistic"</span>,</span>
<span>     pch<span class="op">=</span><span class="fl">20</span>,col<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span>;<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,lty<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6moData_files/figure-html/unnamed-chunk-8-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Same as above, but for proteomics</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">prot_t</span><span class="op">[</span><span class="va">shared_genes</span><span class="op">]</span>,<span class="va">male_8w_prot_ttests</span><span class="op">[</span><span class="st">"tscore"</span>,<span class="va">shared_genes</span><span class="op">]</span>,</span>
<span>     main <span class="op">=</span> <span class="st">"Proteomics DA, males at week 8"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"limma t-score"</span>,ylab <span class="op">=</span> <span class="st">"Simple t-test statistic"</span>,</span>
<span>     pch<span class="op">=</span><span class="fl">20</span>,col<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span>;<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,lty<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6moData_files/figure-html/unnamed-chunk-8-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h2>
<p>For questions, bug reporting, and data requests for this package,
please <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/issues" class="external-link">submit
a new issue</a> and include as many details as possible.</p>
<p>If the concern is related to functions provided in the <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo" class="external-link">MotrpacRatTraining6mo</a>
package, please submit an issue <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/issues" class="external-link">here</a>
instead.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>MoTrPAC is supported by the National Institutes of Health (NIH)
Common Fund through cooperative agreements managed by the National
Institute of Diabetes and Digestive and Kidney Diseases (NIDDK),
National Institute of Arthritis and Musculoskeletal Diseases (NIAMS),
and National Institute on Aging (NIA).</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.4.0                   MotrpacRatTraining6moData_1.7.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] highr_0.10        bslib_0.4.2       compiler_4.2.2    pillar_1.8.1     </span></span>
<span><span class="co">#&gt;  [5] jquerylib_0.1.4   tools_4.2.2       digest_0.6.31     jsonlite_1.8.4   </span></span>
<span><span class="co">#&gt;  [9] evaluate_0.19     memoise_2.0.1     lifecycle_1.0.3   tibble_3.1.8     </span></span>
<span><span class="co">#&gt; [13] gtable_0.3.1      pkgconfig_2.0.3   rlang_1.0.6       DBI_1.1.3        </span></span>
<span><span class="co">#&gt; [17] cli_3.5.0         yaml_2.3.6        pkgdown_2.0.7     xfun_0.36        </span></span>
<span><span class="co">#&gt; [21] fastmap_1.1.0     withr_2.5.0       dplyr_1.0.10      stringr_1.5.0    </span></span>
<span><span class="co">#&gt; [25] knitr_1.41        generics_0.1.3    desc_1.4.2        fs_1.5.2         </span></span>
<span><span class="co">#&gt; [29] vctrs_0.5.1       sass_0.4.4        systemfonts_1.0.4 tidyselect_1.2.0 </span></span>
<span><span class="co">#&gt; [33] rprojroot_2.0.3   grid_4.2.2        glue_1.6.2        R6_2.5.1         </span></span>
<span><span class="co">#&gt; [37] textshaping_0.3.6 fansi_1.0.3       rmarkdown_2.19    farver_2.1.1     </span></span>
<span><span class="co">#&gt; [41] purrr_1.0.0       magrittr_2.0.3    scales_1.2.1      htmltools_0.5.4  </span></span>
<span><span class="co">#&gt; [45] assertthat_0.2.1  colorspace_2.0-3  labeling_0.4.2    ragg_1.2.4       </span></span>
<span><span class="co">#&gt; [49] utf8_1.2.2        stringi_1.7.8     munsell_0.5.0     cachem_1.0.6</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicole Gay, Pierre Jean Beltran, David Amar, David Jimenez-Morales, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
